--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,12 @@
-PROJECT = xa-003
-EXEC = $(PROJECT)
+SHELL = /bin/sh
+PROJECT = xa003midssumer-by-xangle
+EXE = $(PROJECT)
 CPPC = g++ 
 
 # Compile flags ----------------------------------------------------------------
-#FLAGS = -g -Wall
-FLAGS = -O3 -s -Wall
-#FLAGS = -Os -s
+#CXXFLAGS = -g -Wall
+CXXFLAGS += -O3 -flto -Wall -ffast-math -pipe -std=gnu++98 -fabi-version=2
+#CXXFLAGS = -Os -s
 
 # DEBUG ------------------------------------------------------------------------
 #DEBUG = -DDEBUG
@@ -16,11 +17,11 @@
 # GUI --------------------------------------------------------------------------
 GUIOPT = -DUSE_GUI
 GUICODE = base/xdl.o
-GUILIB = `pkg-config --libs gtk+-2.0`
+GUILIB = $(shell pkgconf --libs gtk+-2.0)
 
 # Library and include flags ----------------------------------------------------
-LIB = `sdl-config --libs` -lpng $(GUILIB) -lSDL_mixer
-INC = `sdl-config --cflags` $(GUIOPT) $(DEBUG) $(FPSLOCK) $(NO_STEX)
+LDFLAGS += $(shell pkgconf sdl libpng SDL_mixer --libs) $(GUILIB)
+INC = $(shell sdl-config --cflags) $(GUIOPT) $(DEBUG) $(FPSLOCK) $(NO_STEX)
 
 # Targets ----------------------------------------------------------------------
 OBASE = base/main.o $(GUICODE) base/xdm.o base/xio.o base/xs.o
@@ -30,26 +31,43 @@
 OGFX = $(GFX)cube.o $(GFX)plasma.o $(GFX)font.o $(GFX)anim.o $(GFX)noice.o $(GFX)sort.o
 OBJECTS = $(OBASE) $(OSCENE) $(OGFX)
 
+RM_F = rm -f
+
+#variables d'installation
+INSTALL = install
+INSTALL_DIR     = $(INSTALL) -p -d -o root -g root  -m  755
+INSTALL_PROGRAM = $(INSTALL) -p    -o root -g root  -m  755
+INSTALL_DATA    = $(INSTALL) -p    -o root -g root  -m  644
+
+prefix          = /usr
+exec_prefix     = $(prefix)
+bindir          = $(prefix)/bin
+datarootdir     = $(prefix)/share
+datadir         = $(prefix)/share/xa003midssumer-by-xangle
+
+#parallel compilation if available
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ MAKEFLAGS += -j$(NUMJOBS)
+endif
+
+export
+
 all: $(OBJECTS)
-	$(CPPC) $(FLAGS) $(OBJECTS) -o $(EXEC) $(LIB)
-#	@strip --remove-section=".comment" $(EXEC)
-#	@strip --remove-section=".note" $(EXEC)
-#	@gzexe $(EXEC)
-	@rm -f $(EXEC)~
-	@wc -c $(EXEC)
+	$(CPPC) $(CXXFLAGS) $(OBJECTS) -o $(EXE) $(LDFLAGS)
 
 sstrip: $(OBJECTS)
 #	http://www.muppetlabs.com/~breadbox/software/elfkickers.html
-#	Use FLAGS = -Os -s
-	@$(CPPC) $(FLAGS) $(OBJECTS) -o $(EXEC) $(LIB)
+#	Use CXXFLAGS = -Os -s
+	@$(CPPC) $(CXXFLAGS) $(OBJECTS) -o $(EXEC) $(LDFLAGS)
 	@sstrip $(EXEC)
 	@upx -q $(EXEC)
 	@rm -f $(EXEC)~
 	@wc -c $(EXEC)
 
 upx: $(OBJECTS)
-#	Use FLAGS = -Os -s
-	@$(CPPC) $(FLAGS) $(OBJECTS) -o $(EXEC) $(LIB)
+#	Use CXXFLAGS = -Os -s
+	@$(CPPC) $(CXXFLAGS) $(OBJECTS) -o $(EXEC) $(LDFLAGS)
 	@upx -q $(EXEC)
 	@rm -f $(EXEC)~
 	@wc -c $(EXEC)
@@ -62,39 +80,45 @@
 
 clean:
 	@echo -n Cleaning up...
-	@rm -f $(EXEC) base/*.o scene/*.o gfx/*.o &> /dev/null
+	-@$(RM_F) $(EXE)
+	-@$(RM_F) base/*.o scene/*.o gfx/*.o
 	@echo Done.
 
 # Object dependencys -----------------------------------------------------------
 base/main.o: base/main.cpp
-	$(CPPC) $(FLAGS) -c base/main.cpp -o base/main.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c base/main.cpp -o base/main.o $(INC)
 base/xdl.o: base/xdl.cpp
-	$(CPPC) $(FLAGS) -c base/xdl.cpp -o base/xdl.o $(INC) `pkg-config --cflags gtk+-2.0`
+	$(CPPC) $(CXXFLAGS) -c base/xdl.cpp -o base/xdl.o $(INC) `pkg-config --cflags gtk+-2.0`
 base/xdm.o: base/xdm.cpp
-	$(CPPC) $(FLAGS) -c base/xdm.cpp -o base/xdm.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c base/xdm.cpp -o base/xdm.o $(INC)
 base/xio.o: base/xio.cpp
-	$(CPPC) $(FLAGS) -c base/xio.cpp -o base/xio.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c base/xio.cpp -o base/xio.o $(INC)
 base/xs.o: base/xs.cpp
-	$(CPPC) $(FLAGS) -c base/xs.cpp -o base/xs.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c base/xs.cpp -o base/xs.o $(INC)
 # Scene
 $(SC)intro.o: $(SC)intro.cpp
-	$(CPPC) $(FLAGS) -c $(SC)intro.cpp -o $(SC)intro.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(SC)intro.cpp -o $(SC)intro.o $(INC)
 $(SC)rollingcube.o: $(SC)rollingcube.cpp
-	$(CPPC) $(FLAGS) -c $(SC)rollingcube.cpp -o $(SC)rollingcube.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(SC)rollingcube.cpp -o $(SC)rollingcube.o $(INC)
 $(SC)end.o: $(SC)end.cpp
-	$(CPPC) $(FLAGS) -c $(SC)end.cpp -o $(SC)end.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(SC)end.cpp -o $(SC)end.o $(INC)
 $(SC)greets.o: $(SC)greets.cpp
-	$(CPPC) $(FLAGS) -c $(SC)greets.cpp -o $(SC)greets.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(SC)greets.cpp -o $(SC)greets.o $(INC)
 # Gfx
 $(GFX)cube.o: $(GFX)cube.cpp
-	$(CPPC) $(FLAGS) -c $(GFX)cube.cpp -o $(GFX)cube.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(GFX)cube.cpp -o $(GFX)cube.o $(INC)
 $(GFX)plasma.o: $(GFX)plasma.cpp
-	$(CPPC) $(FLAGS) -c $(GFX)plasma.cpp -o $(GFX)plasma.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(GFX)plasma.cpp -o $(GFX)plasma.o $(INC)
 $(GFX)font.o: $(GFX)font.cpp
-	$(CPPC) $(FLAGS) -c $(GFX)font.cpp -o $(GFX)font.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(GFX)font.cpp -o $(GFX)font.o $(INC)
 $(GFX)anim.o: $(GFX)anim.cpp
-	$(CPPC) $(FLAGS) -c $(GFX)anim.cpp -o $(GFX)anim.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(GFX)anim.cpp -o $(GFX)anim.o $(INC)
 $(GFX)noice.o: $(GFX)noice.cpp
-	$(CPPC) $(FLAGS) -c $(GFX)noice.cpp -o $(GFX)noice.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(GFX)noice.cpp -o $(GFX)noice.o $(INC)
 $(GFX)sort.o: $(GFX)sort.cpp
-	$(CPPC) $(FLAGS) -c $(GFX)sort.cpp -o $(GFX)sort.o $(INC)
+	$(CPPC) $(CXXFLAGS) -c $(GFX)sort.cpp -o $(GFX)sort.o $(INC)
+
+install: $(EXE)
+	$(INSTALL_DIR) $(DESTDIR)$(bindir)
+	-@$(RM_F) $(DESTDIR)$(bindir)/$(EXE)
+	$(INSTALL_PROGRAM) $(EXE) $(DESTDIR)$(bindir)
