--- a/Common.mak
+++ b/Common.mak
@@ -306,6 +306,34 @@
     endif
 endif
 
+### get rid all of that, and use debian stuff
+
+ifeq "$(DEB_HOST_MULTIARCH)" "x86_64-linux-gnu"
+	IMPLICIT_ARCH := x86_64
+	BITS := 64
+endif
+ifeq "$(DEB_HOST_MULTIARCH)" "i386-linux-gnu"
+	IMPLICIT_ARCH := i386
+	BITS := 32
+endif
+ifeq "$(DEB_HOST_MULTIARCH)" "arm-linux-gnueabihf"
+	IMPLICIT_ARCH := ARM
+	BITS := 32
+endif
+ifeq "$(DEB_HOST_MULTIARCH)" "aarch64-linux-gnu"
+	IMPLICIT_ARCH := ARM64
+	BITS := 64
+endif
+
+ifeq "$(DEB_HOST_MULTIARCH)" "aarch64-linux-gnu"
+	IMPLICIT_ARCH := ARM64
+	BITS := 64
+endif
+
+ifeq "$(DEB_HOST_MULTIARCH)" "riscv64-linux-gnu"
+	IMPLICIT_ARCH := RISCV64
+	BITS := 64
+endif
 
 ##### Toggles
 
