--- a/src/main/romloader.cpp
+++ b/src/main/romloader.cpp
@@ -212,4 +212,4 @@
     int size = (int) in.tellg();
     in.close();
     return size; 
-}
\ No newline at end of file
+}
--- a/src/main/engine/omusic.cpp
+++ b/src/main/engine/omusic.cpp
@@ -39,13 +39,13 @@
     if (tilemap == NULL)
     {
         tilemap = new RomLoader();
-        status += tilemap->load_binary("res/tilemap.bin");
+        status += tilemap->load_binary("/usr/share/games/cannonball/res/tilemap.bin");
     }
 
     if (tile_patch == NULL)
     {
         tile_patch = new RomLoader();
-        status += tile_patch->load_binary("res/tilepatch.bin");
+        status += tile_patch->load_binary("/usr/share/games/cannonball/res/tilepatch.bin");
     }
 
     return status == 0;
