--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,24 @@
-CC = gcc
-CFLAGS = -Wall -W -pedantic `sdl-config --cflags`
-CLIBS = -L/usr/X11R6/lib -lGL `sdl-config --libs` -lm -lpng -lSDL_mixer
+#variables de compilation
+SHELL := /bin/sh
+CC    := gcc
+CFLAGS += -Wall -pedantic $(shell pkgconf gl sdl libpng SDL_mixer --cflags) -std=gnu90 -pipe
+LDFLAGS +=  -L/usr/X11R6/lib $(shell pkgconf gl sdl libpng SDL_mixer --libs) -lm
+TARGET   := wayang-by-cyberneticgenetics
+#variable de nettoyage
+RM_F := rm -f
 
-wayang: main.o texture.o mesh3d.o wayang.o movements.o
-	$(CC) $(CLIBS) *.o -o wayang
-	mv wayang ../
+#compilation en paralelle si disponible
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ MAKEFLAGS += -j$(NUMJOBS)
+endif
+
+export
+
+all: $(TARGET)
+
+$(TARGET): main.o texture.o mesh3d.o wayang.o movements.o
+	$(CC) *.o $(LDFLAGS) -o $(TARGET)
 main.o: main.c mesh3d.h texture.h wayang.h
 	$(CC) $(CFLAGS) main.c -c
 texture.o: texture.c texture.h
@@ -18,5 +32,5 @@
 
 clean:
 	@echo Cleaning up...
-	@rm -f *.o *~ core*
+	-@$(RM_F) $(TARGET) *.o *~ core*
 	@echo Done.
