--- a/base/xdl.cpp
+++ b/base/xdl.cpp
@@ -141,13 +141,13 @@
 	GtkWidget *opt = gtk_option_menu_new();
 	GtkWidget *menu = gtk_menu_new();
 	GtkWidget *item;
-	item = makeMenuItem("22050", G_CALLBACK(cbFreq22), NULL);
+	item = makeMenuItem((gchar*)"22050", G_CALLBACK(cbFreq22), NULL);
 	gtk_menu_shell_append(GTK_MENU_SHELL (menu), item);
 	gtk_option_menu_set_menu (GTK_OPTION_MENU (opt), menu);
-	item = makeMenuItem("32000", G_CALLBACK(cbFreq32), NULL);
+	item = makeMenuItem((gchar*)"32000", G_CALLBACK(cbFreq32), NULL);
 	gtk_menu_shell_append(GTK_MENU_SHELL (menu), item);
 	gtk_option_menu_set_menu (GTK_OPTION_MENU (opt), menu);
-	item = makeMenuItem("44100", G_CALLBACK(cbFreq44), NULL);
+	item = makeMenuItem((gchar*)"44100", G_CALLBACK(cbFreq44), NULL);
 	gtk_menu_shell_append(GTK_MENU_SHELL (menu), item);
 	gtk_option_menu_set_menu (GTK_OPTION_MENU (opt), menu);
 
--- a/base/xio.h
+++ b/base/xio.h
@@ -149,7 +149,7 @@
 void loadOne(const char *file);
 void loadAll(const char *dir);
 void add(xio::surface *src, const char *name);
-surface *Texture::get(const char *idstr);
+surface * get(const char *idstr);
 #ifdef DEBUG
 int kbytes;
 #endif
