--- a/capture.h
+++ b/capture.h
@@ -3,6 +3,9 @@
 
 #include <GL/gl.h>
 #include <vector>
+#include <stdio.h>
+using namespace std;
+#include <cstdlib>
 
 class Capture {
 public:
--- a/fontrenderer.cpp
+++ b/fontrenderer.cpp
@@ -1,5 +1,7 @@
 #include "fontrenderer.h"
 #include <string.h>
+#include <GL/gl.h>
+
 
 void FontRenderer::init( std::string filename )
 {
--- a/fontrenderer.h
+++ b/fontrenderer.h
@@ -2,7 +2,7 @@
 #define FONTRENDERER_H
 
 #include <string>
-#include <FTGL/FTGL.h>
+#include <FTGL/ftgl.h>
 #include <FTGL/FTGLTextureFont.h>
 #include <FTGL/FTGLOutlineFont.h>
 #include "Vec3d.h"
--- a/sound.cpp
+++ b/sound.cpp
@@ -3,6 +3,9 @@
 
 #include "fmod.hpp"
 #include "fmod_errors.h"
+using namespace std;
+#include <cstdlib>
+#include <stdio.h>
 
 void ERRCHECK(FMOD_RESULT result)
 {
--- a/texture.h
+++ b/texture.h
@@ -17,7 +17,6 @@
       width=0;
       height=0;
       max=0;
-
       loadmagich( fn );
     }
 
@@ -28,6 +27,7 @@
 
   void loadmagich( std::string fn )
     {
+      Magick::InitializeMagick(NULL);
       Magick::Image mimage( fn );
       width = mimage.size().width();
       height = mimage.size().height();
--- a/view.cpp
+++ b/view.cpp
@@ -21,6 +21,7 @@
 #include "capture.h"
 #include "texture.h"
 #include "textoverlay.h"
+#include <Magick++.h>
 
 using namespace CGLA;
 
@@ -406,6 +407,7 @@
       capturefake = 1;
       printf("enabling capture support!\n");
     }
+
   ssystem.load("DISCODEL.XM");
   smoothCamera = Vec2d(0.0,-2.0);
   glutInit( &argc, argv );
--- a/SConstruct
+++ b/SConstruct
@@ -15,10 +15,9 @@
     incpath = [fmodpath+"/inc/","windows-support/imagemagick/include", "windows-support/freetype/include/", "windows-support/freetype/include/freetype2", "windows-support/ftgl/include", "windows-support/glut"]
     libs += ['CORE_RL_Magick++_', 'CORE_RL_magick_', 'opengl32' , 'glu32', 'X11' ]
 else:
-    fmodpath = "fmodapi40618linux"
-    libpath += [fmodpath+"/api/lib/"]
-    incpath = [fmodpath+"/api/inc/","/usr/include/freetype2"]
-    libs += ["Magick++","Magick", 'GL' ]
+    libpath += ["/usr/lib/x86_64-linux-gnu/"]
+    incpath = ["/usr/include/fmodex/","/usr/include/freetype2"]
+    libs += ["Magick++","Magick", 'GLU','GL' ]
 
 env = Environment( ENV = os.environ, CXXFLAGS=["-O3","-g3"], LINKFLAGS=["-O3","-g3"], CPPPATH=["CGLA/"]+incpath )
 if crosscompile:
--- a/CGLA/CGLA.h
+++ b/CGLA/CGLA.h
@@ -5,6 +5,8 @@
 #pragma warning (disable: 4244 4800)
 #endif
 
+using namespace std;
+#include <cstring>
 #include <cmath>
 #include <cfloat>
 #include <climits>
