--- a/acinclude.m4
+++ b/acinclude.m4
@@ -27,11 +27,6 @@
       fi
     fi
 
-    if test -x vs/sdl2/linux-host/bin/sdl2-config ; then
-      SDL2_CONFIG=vs/sdl2/linux-host/bin/sdl2-config
-      PATH=vs/sdl2/linux-host/bin:$PATH
-    fi
-
     manual_sdl2config="$SDL2_CONFIG"
 
     AC_PATH_PROG(SDL2_CONFIG, sdl2-config, no)
--- a/configure.ac
+++ b/configure.ac
@@ -479,10 +479,10 @@
 pwd=`realpath $srcdir`
 if [[ -z "$pwd" ]]; then pwd=`pwd`; fi
 
-CFLAGS="$CFLAGS -I$pwd -I$pwd/vs/sdlnet/linux-host/include -I$pwd/vs/sdlnet/linux-host/include/SDL"
-LDFLAGS="$LDFLAGS -L$pwd/vs/sdlnet/linux-host/lib"
-CPPFLAGS="$CPPFLAGS -I$pwd -I$pwd/vs/sdlnet/linux-host/include -I$pwd/vs/sdlnet/linux-host/include/SDL"
-CXXFLAGS="$CXXFLAGS -I$pwd -I$pwd/vs/sdlnet/linux-host/include -I$pwd/vs/sdlnet/linux-host/include/SDL"
+CFLAGS="$CFLAGS -I$pwd"
+LDFLAGS="$LDFLAGS"
+CPPFLAGS="$CPPFLAGS -I$pwd"
+CXXFLAGS="$CXXFLAGS -I$pwd"
 
 if test x$enable_emscripten != xyes; then
 dnl Some target detection and actions for them
