--- a/src/main/sdl/audio.cpp
+++ b/src/main/sdl/audio.cpp
@@ -61,7 +61,7 @@
     {
         if(SDL_Init(SDL_INIT_AUDIO) == -1) 
         {
-            std::cout << "Error initalizing audio: " << SDL_GetError() << std::endl;
+            std::cout << "Error initializing audio: " << SDL_GetError() << std::endl;
             return;
         }
 
@@ -77,13 +77,13 @@
 
         if (SDL_OpenAudio(&desired, &obtained) == -1)
         {
-            std::cout << "Error initalizing audio: " << SDL_GetError() << std::endl;
+            std::cout << "Error initializing audio: " << SDL_GetError() << std::endl;
             return;
         }
 
         if (desired.samples != obtained.samples)
         {
-            std::cout << "Error initalizing audio: sample rate not supported." << std::endl;
+            std::cout << "Error initializing audio: sample rate not supported." << std::endl;
             return;
         }
 
@@ -425,4 +425,4 @@
     callbacktick = SDL_GetTicks();
 }
 
-#endif
\ No newline at end of file
+#endif
--- a/src/main/sdl2/audio.cpp
+++ b/src/main/sdl2/audio.cpp
@@ -75,11 +75,11 @@
 
 	    if (SDL_InitSubSystem(SDL_INIT_AUDIO)!=0) {
 
-		std::cout << "Error initalizing audio subsystem: " << SDL_GetError() << std::endl;
+		std::cout << "Error initializing audio subsystem: " << SDL_GetError() << std::endl;
 	    }
 
 	    if (SDL_AudioInit("alsa")!=0) {
-		std::cout << "Error initalizing audio using ALSA: " << SDL_GetError() << std::endl;
+		std::cout << "Error initializing audio using ALSA: " << SDL_GetError() << std::endl;
 		return;
 	    }
 
@@ -87,7 +87,7 @@
 	else {
 	    if(SDL_Init(SDL_INIT_AUDIO) == -1) 
 	    {
-		std::cout << "Error initalizing audio: " << SDL_GetError() << std::endl;
+		std::cout << "Error initializing audio: " << SDL_GetError() << std::endl;
 		return;
 	    }		
 	}
@@ -111,7 +111,7 @@
         }
 
         if (desired.samples != obtained.samples) {
-            std::cout << "Error initalizing audio: number of samples not supported." << std::endl
+            std::cout << "Error initializing audio: number of samples not supported." << std::endl
                       << "Please compare desired vs obtained. Look at what audio driver SDL2 is using." << std::endl;
 	    return;
 	}
