--- a/main.cpp
+++ b/main.cpp
@@ -31,7 +31,7 @@
 	DemoSystem* pDemoSystem;
 	try {
 #ifdef NOFILE
-		pDemoSystem = DemoSystem::Instance(argc, argv, "../demo.xpk", "demo.xml");
+		pDemoSystem = DemoSystem::Instance(argc, argv, "/usr/share/xa005spike-by-xangle/demo.xpk", "demo.xml");
 		//pDemoSystem = DemoSystem::Instance(argc, argv, "demo.xpk", "demo.xml");
 		//pDemoSystem = DemoSystem::Instance(argc, argv, argv[0], "demo.xml");
 #else
