--- a/lib3d/data/lzcompress.cpp
+++ b/lib3d/data/lzcompress.cpp
@@ -2239,7 +2239,7 @@
 #ifdef API3D_SDL2
 #include "../../zlib/zlib.h"
 #else
-#include "../zlib/zlib.h"
+#include <zlib.h>
 #endif
 #endif
 #else
--- a/lib3d/data/png.cpp
+++ b/lib3d/data/png.cpp
@@ -46,7 +46,7 @@
 #ifdef API3D_SDL2
 #include "../../zlib/zlib.h"
 #else
-#include "../zlib/zlib.h"
+#include <zlib.h>
 #endif
 #endif
 #else
--- a/ned/jpeg2000/jpeg2000.cpp
+++ b/ned/jpeg2000/jpeg2000.cpp
@@ -56,7 +56,7 @@
 #include "../player/file_io_zip.h"
 #endif
 
-#include "../../libjp2/openjpeg.h"
+#include <openjpeg.h>
 
 #if defined(WIN32)||defined(API3D_SDL2)
 #pragma warning(disable:4005)
--- a/Player/copypaste.hpp
+++ b/Player/copypaste.hpp
@@ -19,7 +19,7 @@
 char *signCP="DEADDEERCOPY";
 
 #if defined(WIN32)||defined(LINUX)
-#include "../zlib/zlib.h"
+#include <zlib.h>
 #else
 #include "zlib.h"
 #endif
@@ -910,4 +910,4 @@
 #endif
 
 }
-#endif
\ No newline at end of file
+#endif
--- a/Player/script_anim.cpp
+++ b/Player/script_anim.cpp
@@ -465,7 +465,7 @@
 #ifdef API3D_SDL2
 #include "../../zlib/zlib.h"
 #else
-#include "../zlib/zlib.h"
+#include <zlib.h>
 #endif
 #endif
 #else
