--- a/player/main.cpp
+++ b/player/main.cpp
@@ -15,6 +15,7 @@
 #include <wx/glcanvas.h>
 #include <wx/clipbrd.h>
 #include <wx/defs.h>
+#include <wx/cmdline.h>
 #include <GL/gl.h>
 #include <GL/glu.h>
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
@@ -200,20 +201,42 @@
 extern float Inches_h;
 extern float DesktopScale;
 extern char		current_path[1024];
+char current_file[1024];
+wxString datapath;
+wxString datafile;
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 //! Function initializes the application and creates
 //! necessary objects.
 bool App::OnInit()
 {    
 	char ss[1024];
-	FILE *ft=fopen("data/player.conf","r");
+
+	wxCmdLineParser parser(argc,argv);
+    parser.AddParam(_T("files"), wxCMD_LINE_VAL_STRING, wxCMD_LINE_PARAM_MULTIPLE | wxCMD_LINE_PARAM_OPTIONAL);
+
+    parser.Parse(false);
+    if (parser.GetParamCount())
+    {
+        datapath=parser.GetParam(0)+"/data";    
+    } else {
+	    datapath="./data";
+    }
+    datafile=datapath+"/player.conf";
+
+    strcpy(current_path,datapath.ToStdString().c_str());
+    strcpy(current_file,datafile.ToStdString().c_str());
+    
+	FILE *ft=fopen(current_file,"r");
+	if (!ft) {
+        wxPrintf("error: cannot open %s\n", current_file);
+        return false;
+    }
+    wxPrintf("opening %s\n", datafile);
 	fgets(ss,1024,ft);
 	fgets(ss,1024,ft);
 	fgets(ss,1024,ft); str_cleanstuff(ss);
 	fclose(ft);
     
-    strcpy(current_path,"./data");
-    
     loop = false;
 	
     if (strcmp(ss,"windowed")==0) FULLSCREEN=false;
@@ -268,6 +291,7 @@
     return true;
 }
 
+
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 void GLCanvas::ReadMouse(wxMouseEvent & event)
 {
--- a/player/prog.cpp
+++ b/player/prog.cpp
@@ -1119,6 +1119,7 @@
 extern bool windowed;
 extern int LOWRESOLUTION;
 extern char FILEPath[4096];
+extern char current_file[1024];
 int getNumCores();
 
 void evident(char *str,float progression)
@@ -1148,7 +1149,10 @@
 
 	srand ( time(NULL) );
     
-	FILE *ft=fopen("data/player.conf","r");
+	FILE *ft=fopen(current_file,"r");
+	if (!ft) {
+		printf("error: cannot open %s\n", current_file);
+    }
 	fgets(FILENAMENED,1024,ft); str_cleanstuff(FILENAMENED);
 	fgets(EXEPATH,1024,ft); str_cleanstuff(EXEPATH);
 	fgets(ss,1024,ft); str_cleanstuff(ss);
