--- a/cool_gfx/gfx_sdl.c
+++ b/cool_gfx/gfx_sdl.c
@@ -88,24 +88,20 @@
 
 int gfx_exitevent(void)
 {
-    int tmp;
-
-    Uint8   *keys;
-
-    SDL_PumpEvents();
-    if(SDL_QuitRequested())
-        return(1);
-
-/*
-    keys=SDL_GetKeyState(&tmp);
-    if(keys[SDLK_ESCAPE])
-        return(1);
-*/
-
-    if(SDL_GetMouseState(NULL,NULL)&SDL_BUTTON(1))
-        return(1);
-
-    return(0);
+	SDL_Event e;
+	while(SDL_PollEvent(&e))
+	    {
+		if (e.type == SDL_QUIT) {
+		    return(1);
+		}
+		if (e.type == SDL_KEYDOWN) {
+		    switch (e.key.keysym.sym) {
+			case SDLK_ESCAPE: return(1); break;
+			case SDLK_q: return(1); break;
+		    }
+		}
+	    }
+	return(0);
 }
 
 void gfx_close(void)
