--- a/soh/include/functions.h
+++ b/soh/include/functions.h
@@ -73,7 +73,7 @@
 void osSyncPrintfUnused(const char* fmt, ...);
 //void osSyncPrintf(const char* fmt, ...);
 void rmonPrintf(const char* fmt, ...);
-void* is_proutSyncPrintf(void* arg, const char* str, u32 count);
+int is_proutSyncPrintf(void* arg, const char* str, u32 count);
 void func_80002384(const char* exp, const char* file, u32 line);
 OSPiHandle* osDriveRomInit(void);
 void StackCheck_Init(StackEntry* entry, void* stackTop, void* stackBottom, u32 initValue, s32 minSpace,
@@ -140,7 +140,7 @@
 void osSetThreadPri(OSThread* thread, OSPri pri);
 OSPri osGetThreadPri(OSThread* thread);
 s32 __osEPiRawReadIo(OSPiHandle* handle, u32 devAddr, u32* data);
-void osViSwapBuffer(void* vaddr);
+void osViSwapBuffer(int vaddr);
 s32 __osEPiRawStartDma(OSPiHandle* handle, s32 direction, u32 cartAddr, void* dramAddr, size_t size);
 void __osTimerServicesInit(void);
 void __osTimerInterrupt(void);
--- a/soh/src/boot/is_debug.c
+++ b/soh/src/boot/is_debug.c
@@ -34,7 +34,7 @@
     va_end(args);
 }
 
-void* is_proutSyncPrintf(void* arg, const char* str, u32 count) {
+int is_proutSyncPrintf(void* arg, const char* str, u32 count) {
     u32 data;
     s32 pos;
     s32 start;
