--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-ARCH = 
+ARCH =
 #win32
 #osx-x86
 #osx-ppc
@@ -8,12 +8,33 @@
 RANLIB = ranlib
 AR = ar
 MAKE = make
-TARGET = boy
+TARGET = fit039boy-by-fit
 
-CFLAGS = -Wall -O2 -ffast-math `sdl-config --cflags`
-LDFLAGS = `sdl-config --libs` -lGL -lm
+CFLAGS += -Wall $(shell pkgconf sdl gl --cflags) -pipe -std=gnu90
+LDFLAGS += $(shell pkgconf sdl gl --libs) -lm
 OBJ = main.o gl_render.o wavefront.o tga.o cgm.o cgm_gl.o writer.o \
-catmull-rom.o stripe.o stiks.o houses.o vecmat.o 
+catmull-rom.o stripe.o stiks.o houses.o vecmat.o
+
+#variable de nettoyage
+RM_F := rm -f
+#variables d'instalation
+INSTALL := install
+INSTALL_DIR     := $(INSTALL) -p -d -o root -g root  -m  755
+INSTALL_PROGRAM := $(INSTALL) -p    -o root -g root  -m  755
+
+prefix          := /usr
+exec_prefix     := $(prefix)
+bindir          := $(prefix)/bin
+datarootdir     := $(prefix)/share
+datadir         := $(prefix)/share/fit039boy-by-fit
+
+#compilation en paralelle si disponible
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ MAKEFLAGS += -j$(NUMJOBS)
+endif
+
+export
 
 # MinGW cross-compile on Linux
 ifeq ($(ARCH),win32)
@@ -51,11 +72,10 @@
 
 $(TARGET): $(OBJ)
 	$(CC) -o $(TARGET) $(OBJ) $(LDFLAGS)
-	strip $(TARGET)
 
 mpglib/libmpg.a:
-	cd mpglib && $(MAKE)
+	cd mpglib && $(MAKE) $(shell dpkg-buildflags --export=cmdline)
 
 clean:
 	-(cd mpglib && $(MAKE) clean)
-	rm *.o $(TARGET) *~
+	-@$(RM_F) *.o $(TARGET) *~
--- a/mpglib/Makefile
+++ b/mpglib/Makefile
@@ -1,5 +1,6 @@
+CFLAGS += -DLAYER1 -DLAYER2 -DLAYER3 -pipe -std=gnu90
 
-CFLAGS += -DLAYER1 -DLAYER2 -DLAYER3
+PLUGIN := --plugin=$(shell gcc --print-file-name=liblto_plugin.so)
 
 all: libmpg.a
 
@@ -7,12 +8,12 @@
 
 libmpg.a: layer1.o layer2.o common.o dct64_i386.o decode_i386.o layer3.o \
 	tabinit.o interface.o
-	$(AR) rc libmpg.a \
+	$(AR) rc $(PLUGIN) libmpg.a \
 		common.o dct64_i386.o decode_i386.o layer3.o \
 		tabinit.o interface.o layer2.o layer1.o
-	$(RANLIB) libmpg.a
+	$(RANLIB) $(PLUGIN) libmpg.a
 
 # ranlib might not be needed on your system
 
 clean:
-	rm *.o libmpg.a
+	-@$(RM_F) *.o libmpg.a
