--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,7 @@
 
  $(TARGET): main.cc xmodelib.cc texturelib.cpp idscroller.cc timerlib.cpp
 	@echo Compiling Insolit Dust BoRnTrO
-	g++ -o  $(TARGET) $(CXXFLAGS) main.cc xmodelib.cc texturelib.cpp idscroller.cc timerlib.cpp $(shell pkgconf sdl SDL_mixer --libs)
+	g++ -o  $(TARGET) $(CXXFLAGS) main.cc xmodelib.cc texturelib.cpp idscroller.cc timerlib.cpp $(shell pkgconf sdl2 SDL2_mixer --libs) -g
 
 	@echo Build finished. Run ./borntro
 
--- a/main.cc
+++ b/main.cc
@@ -1,5 +1,5 @@
-#include <SDL/SDL.h>
-#include <SDL/SDL_mixer.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_mixer.h>
 #include <math.h>
 #include <unistd.h>
 
@@ -30,8 +30,6 @@
 	Video = new VgaXMode(FULLSCREEN);
 	DBuf = Video->GetVideoMem();
 
-	SDL_WM_SetCaption("InsolitDust BoRnTrO 2012", "borntro");
-
 	Font = new IDScroller("/usr/share/borntro-by-insolitdust/ID_Font_002_32x31.pcx", Video, 32, 31);
 
 	Font->YPosScroller = 208;
--- a/xmodelib.cc
+++ b/xmodelib.cc
@@ -40,36 +40,36 @@
 
 VgaXMode::VgaXMode(char FullScreen)
 {
-	SDL_VideoInfo vinfo;
 	Uint32 vmodeflags = 0;
 	int i;
 
-	//Set the best resolution mode possible
-
-	vmodeflags = SDL_DOUBLEBUF;
-
-	vinfo = *SDL_GetVideoInfo();
-
-	if (vinfo.hw_available == 0)
-		vmodeflags |= SDL_SWSURFACE;
-	else
-	if (vinfo.blit_sw == 0)
-		vmodeflags |= SDL_HWSURFACE;
-	else
-		vmodeflags |= SDL_SWSURFACE;
-
 	//Enable fullscreen mode if is required
 
 	if (FullScreen)
 	{
-		vmodeflags |= SDL_FULLSCREEN;
+		vmodeflags |= SDL_WINDOW_FULLSCREEN;
 
 		SDL_ShowCursor(0);
 	}
 
+	// Initalize Color Masks.
+	#if SDL_BYTEORDER == SDL_BIG_ENDIAN
+	    redMask   = 0xff000000;
+	    greenMask = 0x00ff0000;
+	    blueMask  = 0x0000ff00;
+	    alphaMask = 0x000000ff;
+	#else
+	    redMask   = 0x000000ff;
+	    greenMask = 0x0000ff00;
+	    blueMask  = 0x00ff0000;
+	    alphaMask = 0xff000000;
+	#endif
+
 	//Create the 640x480x256 main "window"
+	sdlWindow = SDL_CreateWindow("InsolitDust BoRnTrO 2012",SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED,640,480,vmodeflags);
+	sdlRenderer = SDL_CreateRenderer(sdlWindow, -1, SDL_RENDERER_TARGETTEXTURE);
 
-	Video_Surface = SDL_SetVideoMode(640, 480, 8, vmodeflags);
+	Video_Surface = SDL_CreateRGBSurface(vmodeflags,640, 480, 32, redMask, greenMask, blueMask, alphaMask);
 
 	if (Video_Surface == NULL)
 		exit(-1);
@@ -81,6 +81,9 @@
 	if (Video_Memory == NULL)
 		exit(-1);
 
+	sdlTexture = SDL_CreateTexture(sdlRenderer, SDL_PIXELFORMAT_ARGB8888,SDL_TEXTUREACCESS_STREAMING,Video_Surface->w, Video_Surface->h);
+	SDL_SetTextureBlendMode(sdlTexture , SDL_BLENDMODE_BLEND);
+
 	//Build then Y offset table for speed up video memory read/write
 
 	PixTab = (unsigned int *)malloc(sizeof(unsigned int) * 240);
@@ -112,12 +115,19 @@
 	SDL_Rect srcrect = {0, 0, 320, 240};
 	SDL_Rect dstrect = {0, 0, 640, 480};
 
-	SDL_SoftStretch(Video_Memory, &srcrect, Video_Surface, &dstrect);
+	//SDL_SoftStretch(Video_Memory, &srcrect, Video_Surface, &dstrect);
 
 	if (SDL_MUSTLOCK(Video_Surface))
 		SDL_UnlockSurface(Video_Surface);
 	else
-		SDL_UpdateRect(Video_Surface, 0, 0, 0, 0);
+		//SDL_UpdateRect(Video_Surface, 0, 0, 0, 0);
+		SDL_BlitScaled(Video_Memory, &srcrect, Video_Surface, &dstrect);
+
+		SDL_UpdateTexture(sdlTexture, &dstrect, Video_Surface->pixels, Video_Surface->pitch);
+		//SDL_UpdateTexture(sdlTexture, NULL, Video_Surface->pixels, Video_Surface->pitch);
+		SDL_RenderClear(sdlRenderer);
+		SDL_RenderCopy(sdlRenderer, sdlTexture, NULL, NULL);
+		SDL_RenderPresent(sdlRenderer);
 }
 
 void VgaXMode::PutPixel(int x, int y, unsigned char col)
@@ -228,7 +238,16 @@
 {
 	//Enabled palette
 
-	SDL_SetColors(Video_Surface, Palette, 0, 256);
+	//SDL_SetColors(Video_Surface, Palette, 0, 256);
+	SDL_SetPaletteColors(Video_Surface->format->palette, Palette, 0, 256);
+	//SDL_UpdateRect(Video_Surface, 0, 0, 0, 0);
+	SDL_BlitSurface(Video_Memory, NULL, Video_Surface, NULL);
+
+	//SDL_UpdateTexture(sdlTexture, &dstrect, Video_Surface->pixels, Video_Surface->pitch);
+	SDL_UpdateTexture(sdlTexture, NULL, Video_Surface->pixels, Video_Surface->pitch);
+	SDL_RenderClear(sdlRenderer);
+	SDL_RenderCopy(sdlRenderer, sdlTexture, NULL, NULL);
+	SDL_RenderPresent(sdlRenderer);
 }
 
 SDL_Color *VgaXMode::GetPalette(void)
@@ -402,7 +421,8 @@
 		return 1;
 	}
 
-	SDL_SetColors(Video_Surface, FadePalette, 0, 256);
+	//SDL_SetColors(Video_Surface, FadePalette, 0, 256);
+	SDL_SetPaletteColors(Video_Surface->format->palette, FadePalette, 0, 256);
 
 	return 0;
 }
@@ -465,7 +485,8 @@
 		return 1;
 	}
 
-	SDL_SetColors(Video_Surface, Palette, 0, 256);
+	//SDL_SetColors(Video_Surface, Palette, 0, 256);
+	SDL_SetPaletteColors(Video_Surface->format->palette, Palette, 0, 256);
 
 	return 0;
 }
@@ -473,4 +494,4 @@
 void VgaXMode::ResetFade(void)
 {
 	NewFade = 0;
-}
\ No newline at end of file
+}
--- a/xmodelib.h
+++ b/xmodelib.h
@@ -31,11 +31,12 @@
 #ifndef XMODELIB_H
 #define XMODELIB_H
 
-#include <SDL/SDL.h>
+#include <SDL2/SDL.h>
 
 class VgaXMode
 {
 private:
+Uint32 redMask, greenMask, blueMask, alphaMask;
 SDL_Surface *Video_Surface;                                     //Primary video surface
 SDL_Surface *Video_Memory;                                      //320x240 surface
 unsigned int *PixTab;                                           //Precalculated Y offset table
@@ -45,6 +46,11 @@
 char NewFade;                               //Uset to reset new fade effect
 
 public:
+
+SDL_Window *sdlWindow;
+SDL_Renderer *sdlRenderer;
+SDL_Texture *sdlTexture;
+
 VgaXMode(char FullScreen);                                      //Constructor, FullScreen = 1 to go in fullscreen mode
 ~VgaXMode();                                                            //Destructor
 
