took system zlib & expat, instead of local one
--- a/libhxcfe/build/Makefile
+++ b/libhxcfe/build/Makefile
@@ -45,7 +45,7 @@
 
 ifeq ($(TARGET), Linux)
 	CFLAGS += -fPIC -Wl,-Map,foo.map
-	LDFLAGS += -lc -lm  -ldl
+	LDFLAGS += -lminizip -llz4 -lz -lexpat -ldl -lspng
 endif
 
 ifeq ($(TARGET), mingw32)
@@ -153,8 +153,8 @@
 		trackeditor.o sector_extractor.o display_track.o fluxStreamAnalyzer.o streamConvert.o script_exec.o env.o
 
 LIBS_OBJS = $(BASE) $(FS_SUPPORT) $(ADFLIBOBJ) $(ENCODING_SUPPORT) $(TRACK_FORMATS_SUPPORT) $(DISKLOADERPLUGINS) \
-			$(DISKWRITERPLUGINS) $(XDMSLIBOBJ) $(ZLIBOBJ) $(MINIZIPOBJ) $(LZ4OBJ) \
-			$(XML_DISK) $(XML_PARSER_EXPAT) $(FATIOLIB) $(LIBSPNG)
+			$(DISKWRITERPLUGINS) $(XDMSLIBOBJ) \
+			$(XML_DISK) $(FATIOLIB)
 
 all: SCRIPT_FILE $(BASEDIR)/licensetxt.h XML_FILES BMP_FILES $(EXEC)
 
--- a/libhxcfe/sources/loaders/adz_loader/adz_loader.c
+++ b/libhxcfe/sources/loaders/adz_loader/adz_loader.c
@@ -61,7 +61,7 @@
 
 #include "loaders/common/raw_amiga.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
+#include <zlib.h>
 
 #include "libhxcadaptor.h"
 
--- a/libhxcfe/sources/loaders/adz_loader/adz_writer.c
+++ b/libhxcfe/sources/loaders/adz_loader/adz_writer.c
@@ -31,7 +31,7 @@
 
 #include "types.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
+#include <zlib.h>
 
 #include "internal_libhxcfe.h"
 #include "tracks/track_generator.h"
--- a/libhxcfe/sources/loaders/afi_loader/afi_loader.c
+++ b/libhxcfe/sources/loaders/afi_loader/afi_loader.c
@@ -58,7 +58,7 @@
 #include "afi_format.h"
 #include "afi_loader.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
+#include <zlib.h>
 #include "tracks/crc.h"
 
 #include "libhxcadaptor.h"
--- a/libhxcfe/sources/loaders/afi_loader/afi_writer.c
+++ b/libhxcfe/sources/loaders/afi_loader/afi_writer.c
@@ -37,7 +37,7 @@
 #include "afi_format.h"
 #include "afi_loader.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
+#include <zlib.h>
 
 #include "tracks/crc.h"
 
--- a/libhxcfe/sources/loaders/imz_loader/imz_loader.c
+++ b/libhxcfe/sources/loaders/imz_loader/imz_loader.c
@@ -59,8 +59,8 @@
 
 #include "imz_loader.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
-#include "thirdpartylibs/zlib/contrib/minizip/unzip.h"
+#include <zlib.h>
+#include <minizip/unzip.h>
 
 #include "loaders/common/raw_iso.h"
 
--- a/libhxcfe/sources/loaders/streamhfe_loader/streamhfe_loader.c
+++ b/libhxcfe/sources/loaders/streamhfe_loader/streamhfe_loader.c
@@ -64,7 +64,7 @@
 
 #include "tracks/luts.h"
 
-#include "thirdpartylibs/lz4/lib/lz4.h"
+#include <lz4.h>
 
 #include "misc/env.h"
 #include "misc/script_exec.h"
--- a/libhxcfe/sources/loaders/streamhfe_loader/streamhfe_writer.c
+++ b/libhxcfe/sources/loaders/streamhfe_loader/streamhfe_writer.c
@@ -63,7 +63,7 @@
 
 #include "tracks/luts.h"
 
-#include "thirdpartylibs/lz4/lib/lz4.h"
+#include <lz4.h>
 
 #define SHIFTER_FREQ 25000000
 
--- a/libhxcfe/sources/loaders/hxcstream_loader/hxcstream_loader.c
+++ b/libhxcfe/sources/loaders/hxcstream_loader/hxcstream_loader.c
@@ -66,7 +66,7 @@
 
 #include "libhxcadaptor.h"
 
-#include "thirdpartylibs/lz4/lib/lz4.h"
+#include <lz4.h>
 
 #include "misc/env.h"
 #include "misc/script_exec.h"
--- a/libhxcfe/sources/loaders/hxcstream_loader/hxcstream.c
+++ b/libhxcfe/sources/loaders/hxcstream_loader/hxcstream.c
@@ -53,7 +53,7 @@
 #include "libhxcfe.h"
 
 #include "tracks/std_crc32.h"
-#include "thirdpartylibs/lz4/lib/lz4.h"
+#include <lz4.h>
 
 #include "floppy_loader.h"
 #include "floppy_utils.h"
--- a/libhxcfe/sources/loaders/bmp_loader/bmp_writer.c
+++ b/libhxcfe/sources/loaders/bmp_loader/bmp_writer.c
@@ -43,7 +43,7 @@
 #include "misc/env.h"
 
 #define SPNG_STATIC 1
-#include "thirdpartylibs/libspng/spng.h"
+#include "spng.h"
 
 static int32_t progress_callback( uint32_t current, uint32_t total, void * td, void * user )
 {
