--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,8 +21,8 @@
 # Gets JUCE from GIT, via CPM.
 CPMAddPackage(
         NAME JUCE
-        GITHUB_REPOSITORY juce-framework/JUCE
-        GIT_TAG 7.0.12)
+        SOURCE_DIR ${CMAKE_SOURCE_DIR}/JUCE
+        JUCE_WEB_BROWSER=0)
 
 add_subdirectory(binaryResources)
 add_subdirectory(source)
@@ -94,4 +94,4 @@
 #target_compile_options(ThirdParty PRIVATE ${NO_WARNING_FLAGS})     # This actually doesn't work. Fallback on pragmas...
 
 # Packaging via CPack.
-include(cpack/CPackOptions.cmake)
+#include(cpack/CPackOptions.cmake)
