--- a/JUCE/modules/juce_audio_plugin_client/detail/juce_CheckSettingMacros.h
+++ b/JUCE/modules/juce_audio_plugin_client/detail/juce_CheckSettingMacros.h
@@ -23,6 +23,30 @@
   ==============================================================================
 */
 
+/* Debian specific overrides */
+#if JucePlugin_Build_LV2 && ! defined (JucePlugin_LV2URI)
+# warning JUCE/Debian using default LV2URI
+# define JucePlugin_LV2URI               JucePlugin_Name
+#endif
+
+#if JucePlugin_Build_AU
+# warning JUCE/Debian disabled AU support
+# undef JucePlugin_Build_AU
+# define JucePlugin_Build_AU 0
+#endif
+
+#if JucePlugin_Build_RTAS
+# warning JUCE/Debian disabled RTAS support
+# undef JucePlugin_Build_RTAS
+# define JucePlugin_Build_RTAS 0
+#endif
+
+#if JucePlugin_Build_AAX
+# warning JUCE/Debian disabled AAX support
+# undef JucePlugin_Build_AAX
+# define JucePlugin_Build_AAX 0
+#endif
+
 // The following checks should cause a compile error if you've forgotten to
 // define all your plugin settings properly..
 
--- a/JUCE/modules/juce_core/juce_core.h
+++ b/JUCE/modules/juce_core/juce_core.h
@@ -65,6 +65,37 @@
 
 #include "system/juce_TargetPlatform.h"
 
+/** Debian specific constants
+  - Debian, we don't use embedded libraries
+*/
+#ifdef JUCE_INCLUDE_ZLIB_CODE
+# undef JUCE_INCLUDE_ZLIB_CODE
+#endif
+#define JUCE_INCLUDE_ZLIB_CODE 0
+
+#ifdef JUCE_INCLUDE_FLAC_CODE
+# undef JUCE_INCLUDE_FLAC_CODE
+#endif
+#define JUCE_INCLUDE_FLAC_CODE 0
+
+#ifdef JUCE_INCLUDE_OGGVORBIS_CODE
+# undef JUCE_INCLUDE_OGGVORBIS_CODE
+#endif
+#define JUCE_INCLUDE_OGGVORBIS_CODE 0
+
+#ifdef JUCE_INCLUDE_JPEGLIB_CODE
+# undef JUCE_INCLUDE_JPEGLIB_CODE
+#endif
+#define JUCE_INCLUDE_JPEGLIB_CODE 0
+
+#ifdef JUCE_INCLUDE_PNGLIB_CODE
+# undef JUCE_INCLUDE_PNGLIB_CODE
+#endif
+#define JUCE_INCLUDE_PNGLIB_CODE 0
+/** Debian specifics END
+*/
+
+
 //==============================================================================
 /** Config: JUCE_FORCE_DEBUG
 
@@ -124,7 +155,7 @@
     specify the path where your zlib headers live.
 */
 #ifndef JUCE_INCLUDE_ZLIB_CODE
- #define JUCE_INCLUDE_ZLIB_CODE 1
+ #define JUCE_INCLUDE_ZLIB_CODE 0
 #endif
 
 #ifndef JUCE_ZLIB_INCLUDE_PATH
