--- a/JUCE/extras/Build/juceaide/CMakeLists.txt
+++ b/JUCE/extras/Build/juceaide/CMakeLists.txt
@@ -41,6 +41,7 @@
         JUCE_SILENCE_XCODE_15_LINKER_WARNING=1)
 
     target_link_libraries(juceaide PRIVATE
+	z
         juce::juce_build_tools
         juce::juce_recommended_config_flags
         juce::juce_recommended_lto_flags
--- a/JUCE/extras/Projucer/CMakeLists.txt
+++ b/JUCE/extras/Projucer/CMakeLists.txt
@@ -168,6 +168,7 @@
     ../Build/CMake/juce_LinuxSubprocessHelper.cpp)
 
 target_link_libraries(Projucer PRIVATE
+    z
     ProjucerData
     juce::juce_build_tools
     juce::juce_cryptography
--- a/JUCE/extras/Projucer/Source/ProjectSaving/jucer_ProjectExport_Make.h
+++ b/JUCE/extras/Projucer/Source/ProjectSaving/jucer_ProjectExport_Make.h
@@ -735,6 +735,9 @@
         StringArray result (linuxLibs);
 
         auto libraries = StringArray::fromTokens (getExternalLibrariesString(), ";", "\"'");
+#ifdef DEBIAN_JUCERPROJECT_LIBS
+        libraries.addArray(StringArray::fromTokens (DEBIAN_JUCERPROJECT_LIBS, ";", "\"'"));
+#endif
         libraries.removeEmptyStrings();
 
         for (auto& lib : libraries)
--- a/JUCE/extras/Projucer/Source/ProjectSaving/jucer_ProjectExporter.cpp
+++ b/JUCE/extras/Projucer/Source/ProjectSaving/jucer_ProjectExporter.cpp
@@ -605,6 +605,40 @@
     if (isCurlEnabled (project) && ! isLoadCurlSymbolsLazilyEnabled (project))
         packages.add ("libcurl");
 
+    // on Debian, a number of 3rd-party libs have been stripped away,
+    // so we need to link to the system libraries
+    if (project.getEnabledModules().isModuleEnabled ("juce_core"))
+    {
+        packages.add ("zlib");
+    }
+    if (project.getEnabledModules().isModuleEnabled ("juce_graphics"))
+    {
+        packages.add ("libjpeg");
+        packages.add ("libpng");
+    }
+    if (project.getEnabledModules().isModuleEnabled ("juce_audio_formats"))
+    {
+        if(project.isConfigFlagEnabled ("JUCE_USE_FLAC", true))
+        {
+            packages.add ("flac");
+        }
+        if(project.isConfigFlagEnabled ("JUCE_USE_OGGVORBIS", true))
+        {
+            packages.add ("vorbis");
+            packages.add ("vorbisfile");
+            packages.add ("vorbisenc");
+            packages.add ("ogg");
+        }
+    }
+    if (project.getEnabledModules().isModuleEnabled ("juce_audio_devices"))
+    {
+        if(project.isConfigFlagEnabled ("JUCE_JACK", true))
+        {
+            packages.add ("jack");
+        }
+    }
+
+
     if (isWebBrowserComponentEnabled (project) && type == PackageDependencyType::compile)
     {
         packages.add ("webkit2gtk-4.0");
