--- a/src/origami.c
+++ b/src/origami.c
@@ -6529,12 +6529,17 @@
 	resolutions = g_list_append(resolutions, "400x300");
 	resolutions = g_list_append(resolutions, "512x384");
 	resolutions = g_list_append(resolutions, "640x480");
+        resolutions = g_list_append(resolutions, "960x540");
 	resolutions = g_list_append(resolutions, "800x600");
 	resolutions = g_list_append(resolutions, "960x720");
+        resolutions = g_list_append(resolutions, "1024x576");
 	resolutions = g_list_append(resolutions, "1024x768");
 	resolutions = g_list_append(resolutions, "1152x864");
+        resolutions = g_list_append(resolutions, "1280x720");
 	resolutions = g_list_append(resolutions, "1280x960");
+        resolutions = g_list_append(resolutions, "1600x900");
 	resolutions = g_list_append(resolutions, "1600x1200");
+        resolutions = g_list_append(resolutions, "1920x1080");
 
 	// Tee ikkuna
 	window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
@@ -6860,7 +6865,7 @@
 	sel_res = 4;
 	sel_fractal_res = 2;
 
-	screen = SDL_SetVideoMode(640, 480, screen_b, SDL_GFX_FLAGS);
+	screen = SDL_SetVideoMode(848, 480, screen_b, SDL_GFX_FLAGS);
 	SDL_WM_SetCaption("Origami options", NULL);
 
 	blend_mode(1);
