--- a/Makefile
+++ b/Makefile
@@ -3,8 +3,8 @@
 SHELL := /bin/sh
 CC    := gcc
 # Try -Os -O1 -O2 -O3
-CFLAGS += -Wall -std=gnu90 -pipe $(shell pkgconf --cflags sdl gl )
-LDFLAGS += $(shell pkgconf --libs sdl gl ) -lm
+CFLAGS += -Wall -std=gnu90 -pipe $(shell pkgconf --cflags sdl2 gl )
+LDFLAGS += $(shell pkgconf --libs sdl2 gl ) -lm
 TARGET := derivations-by-zipola
 
 #variables d’installation
--- a/derivations.c
+++ b/derivations.c
@@ -3,7 +3,7 @@
  * by zipola
  */
 
-#include <SDL/SDL.h>
+#include <SDL2/SDL.h>
 #include <stdlib.h>
 
 #include "gfx.h"
@@ -24,6 +24,8 @@
 	init_snd();
 	sequencer_loop();
 
+	SDL_GL_DeleteContext(sdlGlcontext);
+	SDL_DestroyWindow(sdlWindow);
 	SDL_Quit();
 
 	asm ("movl $1,%eax\n"
--- a/gfx.c
+++ b/gfx.c
@@ -5,8 +5,8 @@
  */
 
 #include <math.h>
-#include <SDL/SDL.h>
-#include <SDL/SDL_opengl.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_opengl.h>
 
 #include "gfx.h"
 
@@ -236,8 +236,8 @@
 {
 	SDL_ShowCursor(SDL_DISABLE);
 	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
-	SDL_WM_SetCaption("derivations", NULL);
-	SDL_SetVideoMode(1024, 768, 24, SDL_OPENGL );
+	sdlWindow = SDL_CreateWindow("derivations",SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED,1024,768,SDL_WINDOW_OPENGL);
+	sdlGlcontext = SDL_GL_CreateContext(sdlWindow);
 	//SDL_SetVideoMode(1024, 768, 24, SDL_OPENGL);
 
 	glEnable(GL_DEPTH_TEST);
--- a/gfx.h
+++ b/gfx.h
@@ -33,4 +33,7 @@
 
 static boid boid_array[N_BOIDS];
 
+SDL_Window *sdlWindow;
+SDL_GLContext sdlGlcontext;
+
 #endif
--- a/letters.c
+++ b/letters.c
@@ -4,8 +4,8 @@
  * Curiously fascinating font engine.
  */
 
-#include <SDL/SDL.h>
-#include <SDL/SDL_opengl.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_opengl.h>
 
 /*
    1    left line
--- a/seq.c
+++ b/seq.c
@@ -4,8 +4,8 @@
  * Sequencing functions.
  */
 
-#include <SDL/SDL.h>
-#include <SDL/SDL_opengl.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_opengl.h>
 
 #include "gfx.h"
 #include "seq.h"
@@ -39,7 +39,7 @@
 		}
 	}
 
-	SDL_GL_SwapBuffers();
+	SDL_GL_SwapWindow(sdlWindow);
 
 	// End if the demo length is passed.
 	if (LENGTH <= t) return 1;
--- a/snd.c
+++ b/snd.c
@@ -4,7 +4,7 @@
  * Sound functions.
  */
 
-#include <SDL/SDL.h>
+#include <SDL2/SDL.h>
 #include <math.h>
 #include <stdint.h>
 
