--- a/Makefile
+++ b/Makefile
@@ -4,8 +4,8 @@
 CXX := g++
 PKG_CONFIG := pkg-config
 CFLAGS += -Wall -pipe -Irubber1.0 $(shell $(PKG_CONFIG) --cflags glu)
-CXXFLAGS += -Wall -pipe -Irubber1.0 -std=c++11 $(shell $(PKG_CONFIG) --cflags glu)
-LDFLAGS += /usr/lib/$(shell dpkg-architecture -qDEB_HOST_MULTIARCH)/libbass.so -lglut $(shell $(PKG_CONFIG) --libs glu) -lm
+CXXFLAGS += -Wall -pipe -Irubber1.0 -std=c++11 $(shell $(PKG_CONFIG) --cflags glu gl)
+LDFLAGS += /usr/lib/$(shell dpkg-architecture -qDEB_HOST_MULTIARCH)/libbass.so -lglut $(shell $(PKG_CONFIG) --libs glu gl) -lm
 
 OBJ = test3d.o Kewlmation.o \
       rubber1.0/Rubber.o rubber1.0/Shader.o rubber1.0/Texture.o \
