--- a/gdk/Makefile.am
+++ b/gdk/Makefile.am
@@ -21,6 +21,7 @@
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-release $(LT_RELEASE) \
 	-export-dynamic \
+	-no-undefined -Wl,--no-undefined \
 	@GLIB_DEPLIBS@	\
 	@x_ldflags@	\
 	@x_libs@	\
--- a/gtk/Makefile.am
+++ b/gtk/Makefile.am
@@ -31,6 +31,7 @@
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	-release $(LT_RELEASE) 					\
 	-export-dynamic						\
+	-no-undefined -Wl,--no-undefined			\
 	@GLIB_DEPLIBS@ 						\
 	@x_ldflags@						\
 	@x_libs@ 						\
