--- a/main.c
+++ b/main.c
@@ -3,7 +3,7 @@
    slack and waffle
  */
 #ifdef SDL
-#include <SDL/SDL.h>
+#include <SDL2/SDL.h>
 #endif
 
 #ifndef OS_X
@@ -742,8 +742,9 @@
 	SDL_Init(SDL_INIT_AUDIO|SDL_INIT_VIDEO); //not necessary in Linux
 #endif
 #ifdef SDL
-	SDL_SetVideoMode(w, h, 0, SDL_OPENGL);
-	SDL_WM_SetCaption("kartoonikum by ExHouse & NecroStudios",NULL);
+    SDL_Window *sdlWindow;
+	sdlWindow = SDL_CreateWindow("kartoonikum by ExHouse & NecroStudios",SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED,w,h,SDL_WINDOW_OPENGL|SDL_WINDOW_RESIZABLE);
+	SDL_Renderer *sdlRenderer = SDL_CreateRenderer(sdlWindow, -1, 0);
 #endif
 
 #ifndef OS_X
@@ -1095,7 +1096,7 @@
 
 		glPopMatrix();
 #ifdef SDL
-		SDL_GL_SwapBuffers();
+		SDL_GL_SwapWindow(sdlWindow);
 		SDL_Delay(25);
 
 		SDL_Event event;
@@ -1117,6 +1118,8 @@
 	}
 
 #ifdef SDL
+	SDL_DestroyRenderer(sdlRenderer);
+	SDL_DestroyWindow(sdlWindow);
 	SDL_Quit();
 #endif
 
--- a/Makefile
+++ b/Makefile
@@ -48,11 +48,11 @@
 ifeq ($(PLATFORM), LINUX)
 	CFLAGS += -DSDL
 	ifeq ($(DEBUG),TRUE)
-		CFLAGS += -ggdb -DDEBUG -fno-strict-aliasing -std=c99 $(shell $(PKG_CONFIG) --cflags sdl) $(INCLUDES)
-		LDFLAGS += $(shell $(PKG_CONFIG) --libs sdl gl glu x11) -lm
+		CFLAGS += -ggdb -DDEBUG -fno-strict-aliasing -std=c99 $(shell $(PKG_CONFIG) --cflags sdl2) $(INCLUDES)
+		LDFLAGS += $(shell $(PKG_CONFIG) --libs sdl2 gl glu x11) -lm
 	else
 		CFLAGS += -Wall -pipe -fmodulo-sched -fno-toplevel-reorder -fno-inline -fno-defer-pop -fno-keep-static-consts -fmerge-all-constants -fstrength-reduce  \
-			-fno-builtin -D__X11__ -DSDL $(shell $(PKG_CONFIG) gl glu sdl --libs) -std=c99 $(INCLUDES)
+			-fno-builtin -D__X11__ -DSDL $(shell $(PKG_CONFIG) gl glu sdl2 --libs) -std=c99 $(INCLUDES)
 		CFLAGS += $(ADDFLAGS)
 		LDFLAGS += -nostdlib -nostartfiles -ldl -L/usr/lib/ $(shell $(PKG_CONFIG) x11 --libs)
 	endif
