took system zlib & expat, instead of local one
--- a/libhxcfe/trunk/build/Makefile
+++ b/libhxcfe/trunk/build/Makefile
@@ -15,13 +15,14 @@
 	LDFLAGS= -shared -lasan
 else
 	CFLAGS=-O3 $(INCLUDES) -Wall
-	LDFLAGS= -shared -s
+	LDFLAGS= -shared
 endif
 
 EXEC=libhxcfe.so
 
 ifeq ($(TARGET), Linux)
 CFLAGS += -fPIC -Wl,-Map,foo.map
+LDFLAGS += -lminizip -llz4 -lz -lexpat -ldl
 LDFLAGS += -g -lc -lm  -ldl
 endif
 
@@ -105,8 +106,8 @@
 		trackeditor.o sector_extractor.o display_track.o fluxStreamAnalyzer.o script_exec.o env.o
 
 LIBS_OBJS = $(BASE) $(FS_SUPPORT) $(ADFLIBOBJ) $(ENCODING_SUPPORT) $(TRACK_FORMATS_SUPPORT) $(DISKLOADERPLUGINS) \
-			$(DISKWRITERPLUGINS) $(XDMSLIBOBJ) $(ZLIBOBJ) $(MINIZIPOBJ) $(LZ4OBJ) \
-			$(XML_DISK) $(XML_PARSER_EXPAT) $(FATIOLIB)
+			$(DISKWRITERPLUGINS) $(XDMSLIBOBJ) \
+			$(XML_DISK) $(FATIOLIB)
 
 all: SCRIPT_FILE XML_FILES $(EXEC)
 
--- a/libhxcfe/trunk/sources/loaders/adz_loader/adz_loader.c
+++ b/libhxcfe/trunk/sources/loaders/adz_loader/adz_loader.c
@@ -61,7 +61,7 @@
 
 #include "loaders/common/raw_amiga.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
+#include <zlib.h>
 
 #include "libhxcadaptor.h"
 
--- a/libhxcfe/trunk/sources/loaders/adz_loader/adz_writer.c
+++ b/libhxcfe/trunk/sources/loaders/adz_loader/adz_writer.c
@@ -31,7 +31,7 @@
 
 #include "types.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
+#include <zlib.h>
 
 #include "internal_libhxcfe.h"
 #include "tracks/track_generator.h"
--- a/libhxcfe/trunk/sources/loaders/afi_loader/afi_loader.c
+++ b/libhxcfe/trunk/sources/loaders/afi_loader/afi_loader.c
@@ -58,7 +58,7 @@
 #include "afi_format.h"
 #include "afi_loader.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
+#include <zlib.h>
 #include "tracks/crc.h"
 
 #include "libhxcadaptor.h"
--- a/libhxcfe/trunk/sources/loaders/afi_loader/afi_writer.c
+++ b/libhxcfe/trunk/sources/loaders/afi_loader/afi_writer.c
@@ -37,7 +37,7 @@
 #include "afi_format.h"
 #include "afi_loader.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
+#include <zlib.h>
 
 #include "tracks/crc.h"
 
--- a/libhxcfe/trunk/sources/loaders/imz_loader/imz_loader.c
+++ b/libhxcfe/trunk/sources/loaders/imz_loader/imz_loader.c
@@ -59,8 +59,8 @@
 
 #include "imz_loader.h"
 
-#include "thirdpartylibs/zlib/zlib.h"
-#include "thirdpartylibs/zlib/contrib/minizip/unzip.h"
+#include <zlib.h>
+#include <minizip/unzip.h>
 
 #include "loaders/common/raw_iso.h"
 
--- a/libhxcfe/trunk/sources/loaders/streamhfe_loader/streamhfe_loader.c
+++ b/libhxcfe/trunk/sources/loaders/streamhfe_loader/streamhfe_loader.c
@@ -64,7 +64,7 @@
 
 #include "tracks/luts.h"
 
-#include "thirdpartylibs/lz4/lib/lz4lib.h"
+#include <lz4.h>
 
 #include "misc/env.h"
 #include "misc/script_exec.h"
--- a/libhxcfe/trunk/sources/loaders/streamhfe_loader/streamhfe_writer.c
+++ b/libhxcfe/trunk/sources/loaders/streamhfe_loader/streamhfe_writer.c
@@ -62,7 +62,7 @@
 
 #include "tracks/luts.h"
 
-#include "thirdpartylibs/lz4/lib/lz4lib.h"
+#include <lz4.h>
 
 // 0XXXXXXX  < 128
 // 10XXXXXX XXXXXXXX < 16K
--- a/libhxcfe/trunk/sources/loaders/hxcstream_loader/hxcstream_loader.c
+++ b/libhxcfe/trunk/sources/loaders/hxcstream_loader/hxcstream_loader.c
@@ -66,7 +66,7 @@
 
 #include "libhxcadaptor.h"
 
-#include "thirdpartylibs/lz4/lib/lz4lib.h"
+#include <lz4.h>
 
 #include "misc/env.h"
 #include "misc/script_exec.h"
--- a/libhxcfe/trunk/sources/loaders/hxcstream_loader/hxcstream.c
+++ b/libhxcfe/trunk/sources/loaders/hxcstream_loader/hxcstream.c
@@ -53,7 +53,7 @@
 #include "libhxcfe.h"
 
 #include "tracks/std_crc32.h"
-#include "thirdpartylibs/lz4/lib/lz4lib.h"
+#include <lz4.h>
 
 #include "floppy_loader.h"
 #include "floppy_utils.h"
