--- a/libswirl/hw/arm7/arm7_jit_virt_x86.cpp
+++ b/libswirl/hw/arm7/arm7_jit_virt_x86.cpp
@@ -111,7 +111,7 @@
         VirtualProtect(ICache, ICacheSize, PAGE_EXECUTE_READWRITE, &old);
 #elif HOST_OS == OS_LINUX || HOST_OS == OS_DARWIN
 
-        printf("\n\t ARM7_TCB addr: %p | from: %p | addr here: %p\n", ICache, ARM7_TCB, armt_init);
+       // printf("\n\t ARM7_TCB addr: %p | from: %p | addr here: %p\n", ICache, ARM7_TCB, armt_init);
 
         if (mprotect(ICache, ICacheSize, PROT_EXEC | PROT_READ | PROT_WRITE))
         {
@@ -455,4 +455,4 @@
 Arm7VirtBackend* Arm7VirtBackend::Create(ARM7Backend* arm, Arm7Context* ctx) {
     return new Arm7VirtBackendX86(arm, ctx);
 }
-#endif
\ No newline at end of file
+#endif
