--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,18 @@
 VERSION = 4.44.64
+DEB_HOST_MULTIARCH ?= $(shell dpkg-architecture -qDEB_HOST_MULTIARCH)
+LIBDIR     = api/lib
+HDRDIR     = api/inc
+prefix = /usr
+prefixlibdir = $(prefix)/lib
+DESTLIBDIR = $(prefixlibdir)/${DEB_HOST_MULTIARCH}
+DESTHDRDIR = $(prefix)/include/fmodex
+ifeq (${DEB_HOST_MULTIARCH},x86_64-linux-gnu)
+	SUFFIX = 64
+endif
+RM_F = rm -f
+INSTALL = install
+INSTALL_DIR     = $(INSTALL) -p -d -o root -g root  -m  755
+INSTALL_DATA    = $(INSTALL) -p    -o root -g root  -m  644
 
 all:
 	@echo "Possible targets:"
@@ -44,3 +58,22 @@
 	cd fmoddesignerapi/examples/realtime_tweaking && make ${CPU}
 	cd fmoddesignerapi/examples/simple_event && make ${CPU}
 	@echo "done."
+
+install:
+	@echo "Installing FMOD Ex libraries and headers..."
+	$(INSTALL_DIR) $(DESTDIR)${prefixlibdir}
+	$(INSTALL_DIR) $(DESTDIR)${DESTLIBDIR}
+	cp ${LIBDIR}/libfmodex$(SUFFIX)-${VERSION}.so $(DESTDIR)${DESTLIBDIR}/libfmodex.so
+	cp ${LIBDIR}/libfmodexL$(SUFFIX)-${VERSION}.so $(DESTDIR)${DESTLIBDIR}/libfmodexL.so
+	ldconfig -n $(DESTDIR)${DESTLIBDIR}
+	$(INSTALL_DIR) $(DESTDIR)${DESTHDRDIR}
+	$(INSTALL_DATA) ${HDRDIR}/*.h* $(DESTDIR)${DESTHDRDIR}
+	@echo "done."
+
+uninstall:
+	@echo "Uninstalling FMOD Ex libraries..."
+	-@$(RM_F) $(DESTDIR)${DESTLIBDIR}/libfmodex.so
+	-@$(RM_F) $(DESTDIR)${DESTLIBDIR}/libfmodexL.so
+	ldconfig -n $(DESTDIR)${DESTLIBDIR}
+	-@$(RM_F) $(DESTDIR)${DESTHDRDIR}
+	@echo "done."
