--- a/makefile.linux
+++ /dev/null
@@ -1,27 +0,0 @@
-# ============================================================================
-#  Makefile for Linux version of MiniFMOD test player
-# ============================================================================
-
-# = LIBRARY DEFINITIONS ======================================================
-
-LIBPATH  = ./lib/linux
-LIB      = $(LIBPATH)/libminifmod.a
-SRCPATH  = ./src
-INCPATH  = ./include
-
-# = CORE DEFINITIONS =========================================================
-
-CFLAGS   = -s -Os -c -I$(INCPATH) -D__LINUX__ -Wall 
-LFLAGS   = -s  -L$(LIBPATH) -lminifmod -lpthread -lm 
-
-# = MAKEFILE =================================================================
-
-test : $(LIB) main.o
-	gcc main.o $(LFLAGS) -o test
-
-main.o : main.cpp
-	gcc $(CFLAGS) $<
-
-$(LIB) : 
-	(cd $(SRCPATH); make -f makefile.linux)
-
--- a/src/makefile.linux
+++ /dev/null
@@ -1,56 +0,0 @@
-# ============================================================================
-#  Makefile for Linux version of MiniFMOD
-# ============================================================================
-
-# = LIBRARY DEFINITIONS ======================================================
-
-LIBPATH  = ../lib/linux
-LIBNAME  = libminifmod.a
-LIB      = $(LIBPATH)/$(LIBNAME)
-OBJS     = fmusic.o \
-           fsound.o \
-           system_file.o \
-           mixer_fpu_ramp.o \
-           mixer_fpu_ramp_asm.o \
-           mixer_clipcopy.o \
-           music_formatxm.o 
-
-# = TEST DEFINITIONS =========================================================
-
-TESTPATH = ..
-TESTNAME = test
-TEST     = $(TESTPATH)/$(TESTNAME)
-
-# = CORE DEFINITIONS =========================================================
-
-CFLAGS   = -s -Os -c -D__LINUX__ -D__COMPILE_MINIFMOD__ -Wall 
-NFLAGS   = -f elf -D__LINUX__ 
-LFLAGS   = -s -L$(LIBPATH) -lminifmod -lpthread -lm 
-AFLAGS   = -rcvs
-
-# = MAKEFILE =================================================================
-
-$(LIB) : $(OBJS)
-	ar $(AFLAGS) $(LIB) $(OBJS)
-
-$(TESTNAME) : $(LIB) main.o
-	gcc main.o $(LFLAGS) -o $(TEST)
-
-%.o : %.cpp
-	gcc $(CFLAGS) $<
-
-%.o : %.c
-	gcc $(CFLAGS) $<
-
-%.o : %.asm
-	nasm $(NFLAGS) $<
-
-mixer_fpu_ramp_asm.o : mixer_fpu_ramp_asm.asm mixer_fpu_ramp_asm.inc
-	nasm $(NFLAGS) $<
-
-mixer_fpu_ramp_asm.inc : genasminc
-	./genasminc >mixer_fpu_ramp_asm.inc
-        
-genasminc : genasminc.c
-	gcc $(CFLAGS) -s -O3 genasminc.c -o genasminc.o
-	gcc -s genasminc.o -o genasminc
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,65 @@
+#variables de compilation
+SHELL = /bin/sh
+MAKE = make
+CC = gcc
+MAKE = make
+#variable de nettoyage
+RM_F = rm -f
+# ============================================================================
+#  Makefile for Linux version of MiniFMOD test player
+# ============================================================================
+
+# = LIBRARY DEFINITIONS ======================================================
+
+LIBPATH  = ./lib/linux
+LIB      = $(LIBPATH)/libminifmod.a
+SRCPATH  = ./src
+INCPATH  = ./include
+
+# = CORE DEFINITIONS =========================================================
+
+CFLAGS   += -c -I$(INCPATH) -D__LINUX__ -Wall
+LFLAGS   += -L$(LIBPATH) -lminifmod -lpthread -lm
+TARGET   = test
+
+#variables d'instalation
+INSTALL = install
+INSTALL_DIR     = $(INSTALL) -p -d -o root -g root  -m  755
+INSTALL_LIB = $(INSTALL) -p    -o root -g root  -m  755
+
+prefix          = /usr
+exec_prefix     = $(prefix)
+libdir          = $(prefix)/lib/$(DEB_HOST_MULTIARCH)
+
+# = MAKEFILE =================================================================
+
+#strip program
+ifeq (,$(filter nostrip,$(DEB_BUILD_OPTIONS)))
+ INSTALL_PROGRAM += -s
+endif
+#parallel compilation if available
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ MAKEFLAGS += -j$(NUMJOBS)
+endif
+
+export
+
+
+$(TARGET): $(LIB) main.o
+	$(CC) main.o $(LFLAGS) -o test
+
+main.o : main.cpp
+	$(CC) $(CFLAGS) $<
+
+$(LIB) :
+	(cd $(SRCPATH); $(MAKE) -f Makefile)
+
+clean:
+	-@$(RM_F) *.o $(LIB) $(TARGET) *~
+	(cd $(SRCPATH); $(MAKE) -f Makefile clean)
+
+install: $(TARGET)
+	$(INSTALL_DIR) $(DESTDIR)$(libdir)
+	-@$(RM_F) $(DESTDIR)$(LIB)
+	$(INSTALL_LIB) $(LIB) $(DESTDIR)$(libdir)
--- /dev/null
+++ b/src/Makefile
@@ -0,0 +1,72 @@
+# ============================================================================
+#  Makefile for Linux version of MiniFMOD
+# ============================================================================
+
+#variables de compilation
+SHELL = /bin/sh
+MAKE = make
+CC = gcc
+AR= ar
+NASM = nasm
+MAKE = make
+RANLIB = ranlib
+#variable de nettoyage
+RM_F = rm -f
+
+# = LIBRARY DEFINITIONS ======================================================
+
+LIBPATH  = ../lib/linux
+LIBNAME  = libminifmod.a
+LIB      = $(LIBPATH)/$(LIBNAME)
+OBJS     = fmusic.o \
+           fsound.o \
+           system_file.o \
+           mixer_fpu_ramp.o \
+           mixer_fpu_ramp_asm.o \
+           mixer_clipcopy.o \
+           music_formatxm.o
+
+# = TEST DEFINITIONS =========================================================
+
+TESTPATH = ..
+TESTNAME = test
+TEST     = $(TESTPATH)/$(TESTNAME)
+
+# = CORE DEFINITIONS =========================================================
+
+CFLAGS   += -c -D__LINUX__ -D__COMPILE_MINIFMOD__ -Wall
+NFLAGS   = -f elf -D__LINUX__
+LFLAGS   += -L$(LIBPATH) -lminifmod -lpthread -lm
+AFLAGS   = -rcvs
+
+
+# = MAKEFILE =================================================================
+
+$(LIB) : $(OBJS)
+	$(AR) $(AFLAGS) $(LIB) $(OBJS)
+	$(RANLIB) $(LIB)
+
+$(TESTNAME) : $(LIB) main.o
+	$(CC) main.o $(LFLAGS) -o $(TEST)
+
+%.o : %.cpp
+	$(CC) $(CFLAGS) $<
+
+%.o : %.c
+	$(CC) $(CFLAGS) $<
+
+%.o : %.asm
+	$(NASM) $(NFLAGS) $<
+
+mixer_fpu_ramp_asm.o : mixer_fpu_ramp_asm.asm mixer_fpu_ramp_asm.inc
+	$(NASM) $(NFLAGS) $<
+
+mixer_fpu_ramp_asm.inc : genasminc
+	./genasminc >mixer_fpu_ramp_asm.inc
+
+genasminc : genasminc.c
+	$(CC) $(CFLAGS) genasminc.c -o genasminc.o
+	$(CC) genasminc.o -o genasminc
+
+clean:
+	-@$(RM_F) *.o $(LIB) *~
