# set client ID. Generated own ID from twitch
--- a/src/chatty/Chatty.java
+++ b/src/chatty/Chatty.java
@@ -39,7 +39,7 @@
     /**
      * The Twitch client id of this program.
      */
-    public static final String CLIENT_ID = "spyiu9jqdnfjtwv6l1xjk5zgt8qb91l";
+    public static final String CLIENT_ID = "q1b6bqsamkie9jl3917hsvaq388psra";
     
     /**
      * The redirect URI for getting a token.
