--- a/shell/linux/Makefile
+++ b/shell/linux/Makefile
@@ -251,7 +251,7 @@
 LIBS  += -lpthread
 
 PREFIX ?= /usr/local
-MAN_DIR ?= ${PREFIX}/share/man/man1
+MAN_DIR = ${PREFIX}/share/man/man6
 MENUENTRY_DIR ?= ${PREFIX}/share/applications
 ICON_DIR ?= ${PREFIX}/share/pixmaps
 
@@ -389,37 +389,41 @@
 	$(AS) $(ASFLAGS) $(INCS) $< -o $@
 
 install: $(EXECUTABLE)
-	mkdir -p $(DESTDIR)$(PREFIX)/bin 2>/dev/null || true
+	mkdir -p $(DESTDIR)$(PREFIX)/games 2>/dev/null || true
 	mkdir -p $(DESTDIR)$(PREFIX)/share/reicast/mappings 2>/dev/null || true
 	mkdir -p $(DESTDIR)$(MAN_DIR) 2>/dev/null || true
 	mkdir -p $(DESTDIR)$(MENUENTRY_DIR) 2>/dev/null || true
 	mkdir -p $(DESTDIR)$(ICON_DIR) 2>/dev/null || true
-	install -m755 $(EXECUTABLE) $(DESTDIR)$(PREFIX)/bin/$(EXECUTABLE_NAME)
-	install -m755 tools/reicast-joyconfig.py $(DESTDIR)$(PREFIX)/bin/reicast-joyconfig
+	install -m755 $(EXECUTABLE) $(DESTDIR)$(PREFIX)/games/$(EXECUTABLE_NAME)
+	install -m755 tools/reicast-joyconfig.py $(DESTDIR)$(PREFIX)/games/reicast-joyconfig
 	install -m644 mappings/controller_gcwz.cfg $(DESTDIR)$(PREFIX)/share/reicast/mappings
 	install -m644 mappings/controller_generic.cfg $(DESTDIR)$(PREFIX)/share/reicast/mappings
 	install -m644 mappings/controller_pandora.cfg $(DESTDIR)$(PREFIX)/share/reicast/mappings
 	install -m644 mappings/controller_xboxdrv.cfg $(DESTDIR)$(PREFIX)/share/reicast/mappings
 	install -m644 mappings/controller_xpad.cfg $(DESTDIR)$(PREFIX)/share/reicast/mappings
 	install -m644 mappings/keyboard.cfg $(DESTDIR)$(PREFIX)/share/reicast/mappings
-	install -m644 man/reicast.1 $(DESTDIR)$(MAN_DIR)
-	install -m644 man/reicast-joyconfig.1 $(DESTDIR)$(MAN_DIR)
+	cp man/reicast.1 man/reicast.6
+	install -m644 man/reicast.6 $(DESTDIR)$(MAN_DIR)
+	cp man/reicast-joyconfig.1 man/reicast-joyconfig.6
+	install -m644 man/reicast-joyconfig.6 $(DESTDIR)$(MAN_DIR)
 	install -m644 reicast.desktop $(DESTDIR)$(MENUENTRY_DIR)
 	install -m644 reicast.png $(DESTDIR)$(ICON_DIR)
 
 uninstall:
-	rm -f $(DESTDIR)$(PREFIX)/bin/$(EXECUTABLE_NAME)
-	rm -f $(DESTDIR)$(PREFIX)/bin/reicast-joyconfig
+	rm -f $(DESTDIR)$(PREFIX)/games/$(EXECUTABLE_NAME)
+	rm -f $(DESTDIR)$(PREFIX)/games/reicast-joyconfig
 	rm -f $(DESTDIR)$(PREFIX)/share/reicast/mappings/controller_gcwz.cfg
 	rm -f $(DESTDIR)$(PREFIX)/share/reicast/mappings/controller_generic.cfg
 	rm -f $(DESTDIR)$(PREFIX)/share/reicast/mappings/controller_pandora.cfg
 	rm -f $(DESTDIR)$(PREFIX)/share/reicast/mappings/controller_xboxdrv.cfg
 	rm -f $(DESTDIR)$(PREFIX)/share/reicast/mappings/controller_xpad.cfg
 	rm -f $(DESTDIR)$(PREFIX)/share/reicast/mappings/keyboard.cfg
-	rm -f $(DESTDIR)$(MAN_DIR)/reicast.1
-	rm -f $(DESTDIR)$(MAN_DIR)/reicast-joyconfig.1
+	rm -f $(DESTDIR)$(MAN_DIR)/reicast.6
+	rm -f $(DESTDIR)$(MAN_DIR)/reicast-joyconfig.6
 	rm -f $(DESTDIR)$(MENUENTRY_DIR)/reicast.desktop
 	rm -f $(DESTDIR)$(ICON_DIR)/reicast.png
 
 clean:
 	rm -f $(OBJECTS) $(EXECUTABLE) $(EXECUTABLE_STRIPPED) .map
+	rm -f man/reicast.6
+	rm -f man/reicast-joyconfig.6
--- a/shell/linux/man/reicast-joyconfig.1
+++ b/shell/linux/man/reicast-joyconfig.1
@@ -1,6 +1,6 @@
-.\" reicast.1:
+.\" reicast.6:
 
-.TH  "REICAST-JOYCONFIG" "1" "August 27, 2015" "REICAST-JOYCONFIG" "reicast-joyconfig"
+.TH  "REICAST-JOYCONFIG" "6" "August 27, 2015" "REICAST-JOYCONFIG" "reicast-joyconfig"
 
 .SH NAME
 
@@ -71,4 +71,4 @@
 This is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License (Version 2).
 
 .SH "SEE ALSO"
-\fBreicast\fR(1)
+\fBreicast\fR(6)
--- a/shell/linux/man/reicast.1
+++ b/shell/linux/man/reicast.1
@@ -1,6 +1,6 @@
-.\" reicast.1:
+.\" reicast.6:
 
-.TH  "REICAST" "1" "August 22, 2015" "REICAST" "System Manager's Manual: reicast"
+.TH  "REICAST" "6" "August 22, 2015" "REICAST" "System Manager's Manual: reicast"
 
 .SH NAME
 
@@ -127,4 +127,4 @@
 This is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License (Version 2).
 
 .SH "SEE ALSO"
-\fBreicast-joyconfig\fR(1)
+\fBreicast-joyconfig\fR(6)
