--- a/bluecubes.c
+++ b/bluecubes.c
@@ -3,9 +3,10 @@
 
 #include <stdlib.h>
 #include <math.h>
-#include <SDL/SDL.h>
-#include <SDL/SDL_opengl.h>
-#include <SDL/SDL_audio.h>
+#include <GL/glu.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_opengl.h>
+#include <SDL2/SDL_audio.h>
 
 #include "data.c"
 
@@ -13,7 +14,7 @@
 #define PI 3.1415926535897f
 #define startpos 0//30000
 
-SDL_Surface *screen;
+SDL_Window *screen;
 float ar; /* aspect ratio, width/height */
 Uint32 w, h;
 char quit = 0;
@@ -37,7 +38,7 @@
 	SDL_Event event;
 	Uint32 tlast = 0,
 	       tcurrent, st, argn = 0;
-	Uint32 flags = SDL_OPENGL | SDL_HWPALETTE | SDL_FULLSCREEN;
+	Uint32 flags = SDL_WINDOW_FULLSCREEN;
 	char fr = 0;
 
 	while(++argn < argc) {
@@ -50,7 +51,7 @@
 			default: fr=0; break;
 			}
 		} if(argv[argn][0] == '-' && argv[argn][1] == 'w')
-			flags ^= SDL_FULLSCREEN;
+			flags ^= SDL_WINDOW_FULLSCREEN;
 	}
 
 
@@ -60,15 +61,15 @@
 	SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
 	SDL_GL_SetAttribute(SDL_GL_MULTISAMPLEBUFFERS, 1);
 
-	if((fr == 1 && !(screen = SDL_SetVideoMode(w, h, 0, flags))) ||
-	   (fr == 0&& !((screen = SDL_SetVideoMode(1280, 960, 0, flags)) && (w=1280) && (h=960))
-	    && !((screen = SDL_SetVideoMode(1024, 768, 0, flags)) && (w=1024) && (h=768))
-	    && !((screen = SDL_SetVideoMode( 800, 600, 0, flags)) &&  (w=800) && (h=600)))) {
+	if((fr == 1 && !(screen = SDL_CreateWindow("Blue Cubes by YorHel",SDL_WINDOWPOS_UNDEFINED,SDL_WINDOWPOS_UNDEFINED,w, h,flags))) ||
+	   (fr == 0&& !((screen = SDL_CreateWindow("Blue Cubes by YorHel",SDL_WINDOWPOS_UNDEFINED,SDL_WINDOWPOS_UNDEFINED,1280,960,flags)) && (w=1280) && (h=960))
+	    && !((screen = SDL_CreateWindow("Blue Cubes by YorHel",SDL_WINDOWPOS_UNDEFINED,SDL_WINDOWPOS_UNDEFINED,1024, 768,flags)) && (w=1024) && (h=768))
+	    && !((screen = SDL_CreateWindow("Blue Cubes by YorHel",SDL_WINDOWPOS_UNDEFINED,SDL_WINDOWPOS_UNDEFINED,800, 600,flags)) &&  (w=800) && (h=600)))) {
 		printf("Error opening screen: %s\n", SDL_GetError());
 		return 1;
 	}
+	SDL_Renderer *renderer = SDL_CreateRenderer(screen, -1, 0);
 	ar = (float) w / (float) h;
-	SDL_WM_SetCaption("Blue Cubes by YorHel", "Blue Cubes by YorHel");
 	SDL_ShowCursor(0);
 
 	glEnable(GL_BLEND);
@@ -105,10 +106,12 @@
 
 		NextFrame();
 		glFlush();
-		SDL_GL_SwapBuffers();
+		SDL_GL_SwapWindow(screen);
 		tlast = tcurrent;
 	}
 	closesound();
+	SDL_DestroyRenderer(renderer);
+	SDL_DestroyWindow(screen);
 	SDL_Quit();
 	return 0;
 }
@@ -349,6 +352,3 @@
 	}
 }
 
-
-
-
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,8 @@
 #variables de compilation
 SHELL := /bin/sh
 CC    := gcc
-CFLAGS   += -Wall -pipe -std=c99 -pipe -pedantic $(shell pkgconf glu sdl sdl --cflags)
-LDFLAGS  += $(shell pkgconf glu sdl sdl --libs) -lm -pipe
+CFLAGS   += -Wall -pipe -std=c99 -pipe -pedantic $(shell pkgconf glu sdl2 --cflags)
+LDFLAGS  += $(shell pkgconf glu sdl2 --libs) -lm -pipe
 #variable de nettoyage
 RM_F := rm -f
 #variables d'instalation
