--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 SHELL = /bin/sh
 CC    = gcc
 CFLAGS   += -flto -Wall -ffast-math
-LDFLAGS  += -lpthread -lfmodex-3.75 -lm
+LDFLAGS  += -lpthread -lfmodex -lm
 TARGET   = kebab-by-rebel
 #variable de nettoyage
 RM_F = rm -f
--- a/kebab.c
+++ b/kebab.c
@@ -14,7 +14,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <math.h>
-#include <fmod.h>
+#include <fmodex/fmod.h>
 
 #define RESET           0
 #define BRIGHT          1
@@ -450,6 +450,10 @@
 
 }
 
+FMOD_SYSTEM *sys;
+FMOD_SOUND *fsoundkebab;
+FMOD_SOUND *fsoundknark;
+FMOD_CHANNEL *channel;
 
 int main(void)
 {
@@ -458,33 +462,12 @@
 pthread_t thread[1000];
 char snopp[50];
 char ahh[5000];
-FMUSIC_MODULE *mod = NULL;
-FMUSIC_MODULE *knark = NULL;
 
-    if (FSOUND_GetVersion() < FMOD_VERSION)
-    {
-printf("ARGH! fel version av fmod\n");
-        exit(1);
-    }
-
-    if (!FSOUND_Init(32000, 64, 0))
-    {
-printf("INIT SOUND error\n");
-        exit(1);
-    }
-
-    mod = FMUSIC_LoadSong("kebab.xm");
-    if (!mod)
-    {
-printf("Loadsong error\n");
-        exit(1);
-    }
-    knark = FMUSIC_LoadSong("knark.xm");
-    if (!knark)
-    {
-printf("Loadsong error\n");
-        exit(1);
-    }
+FMOD_System_Create(&sys);
+FMOD_System_Init(sys, 1, FMOD_INIT_NORMAL, NULL);
+FMOD_System_CreateSound(sys, "kebab.xm", FMOD_SOFTWARE | FMOD_CREATESTREAM, 0, &fsoundkebab);
+FMOD_System_CreateSound(sys, "knark.xm", FMOD_SOFTWARE | FMOD_CREATESTREAM, 0, &fsoundknark);
+
 
 
 setvbuf(stdout, NULL, _IONBF, 0);
@@ -621,38 +604,39 @@
 fill(BRIGHT,WHITE,BLUE,' ');
 usleep(500000);
 mutta(9,3,BRIGHT,WHITE,WHITE,buf2);
-FMUSIC_PlaySong(knark);
+FMOD_System_PlaySound(sys, FMOD_CHANNEL_FREE, fsoundknark, 0, &channel);
 usleep(500000);
-FMUSIC_StopSong(knark);
+FMOD_Channel_SetPaused(channel, 1);
 
 
 
 
 fill(BRIGHT,WHITE,BLUE,' ');
 mutta(9,3,BRIGHT,WHITE,WHITE,buf3);
-FMUSIC_PlaySong(knark);
+FMOD_Channel_SetPaused(channel, 0);
 usleep(500000);
-FMUSIC_StopSong(knark);
+FMOD_Channel_SetPaused(channel, 1);
 fill(BRIGHT,WHITE,BLUE,' ');
 mutta(9,3,BRIGHT,WHITE,WHITE,buf4);
-FMUSIC_PlaySong(knark);
+FMOD_Channel_SetPaused(channel, 0);
 usleep(500000);
-FMUSIC_StopSong(knark);
+FMOD_Channel_SetPaused(channel, 1);
 fill(BRIGHT,WHITE,BLUE,' ');
 mutta(9,3,BRIGHT,WHITE,WHITE,buf5);
-FMUSIC_PlaySong(knark);
+FMOD_Channel_SetPaused(channel, 0);
 usleep(500000);
-FMUSIC_StopSong(knark);
+FMOD_Channel_SetPaused(channel, 1);
 fill(BRIGHT,WHITE,BLUE,' ');
 mutta(9,3,BRIGHT,WHITE,WHITE,buf6);
-FMUSIC_PlaySong(knark);
+FMOD_Channel_SetPaused(channel, 0);
 usleep(500000);
-FMUSIC_StopSong(knark);
+FMOD_Channel_SetPaused(channel, 1);
+FMOD_Sound_Release(fsoundknark);
 //fill(BRIGHT,WHITE,BLUE,' ');
 mutta(4,3,BRIGHT,WHITE,WHITE,buf7);
 usleep(2500000);
 
-    FMUSIC_PlaySong(mod);
+	FMOD_System_PlaySound(sys, FMOD_CHANNEL_FREE, fsoundkebab, 0, &channel);
 
 
 for(x=1;x<120;x++)
@@ -789,21 +773,12 @@
 fill(BRIGHT,WHITE,BLUE,' ');
 mutta(20,5,BRIGHT,WHITE,WHITE,ahh);
 
-x = FMUSIC_GetMasterVolume(mod);
-
-
-while(x >= 0)
-{
-FMUSIC_SetMasterVolume(mod, x-=10);
-usleep(100000);
-}
 
 
 
-FMUSIC_FreeSong(mod);
-FSOUND_Close();
-
-
+FMOD_Sound_Release(fsoundkebab);
+FMOD_System_Close(sys);
+FMOD_System_Release(sys);
 
 exit(0);
 
@@ -2123,6 +2098,4 @@
 fill(BRIGHT,WHITE,BLUE,' ');
 mutta(COLUMNS/2-9,LINES/2-4,BRIGHT,WHITE,WHITE,smile);
 usleep(800000);
-}
-
-
+}
\ No newline at end of file
