--- a/share/applications/rubyripper.desktop
+++ b/share/applications/rubyripper.desktop
@@ -1,9 +1,11 @@
 [Desktop Entry]
+Version=1.0
 Type=Application
 Name=Rubyripper
 GenericName=Secure Audio Disc Ripper
 GenericName[nl]=Veilige Audio CD Ripper
+GenericName[fr]=Ripper de CD Audio Sécurisé
 Icon=rubyripper
-Exec=rrip_gui
+Exec=/usr/bin/rrip_gui
 Categories=Audio;AudioVideo;
 Keywords=Audio;Music;CD;Rip;Ripper;Disc;Disk;MP3;FLAC;Vorbis;Ogg;WAV;Transcode;Convert;
