--- a/GNUmakefile
+++ b/GNUmakefile
@@ -901,6 +901,16 @@
     editor \
 
 
+#variables d'instalation
+INSTALL := install
+INSTALL_DIR     := $(INSTALL) -p -d -o root -g root  -m  755
+INSTALL_PROGRAM := $(INSTALL) -p    -o root -g root  -m  755
+
+
+prefix          := /usr
+exec_prefix     := $(prefix)
+bindir          := $(prefix)/games
+
 ifeq ($(PRETTY_OUTPUT),1)
 .SILENT:
 endif
@@ -1123,7 +1133,6 @@
 
 rev: $(engine_obj)/rev.$o
 
-
 ### Compatibility
 
 cleantest: cleankenbuild
