--- /dev/null
+++ b/Makefile
@@ -0,0 +1,32 @@
+#variables de compilation
+SHELL = /bin/sh
+CXX   = g++
+CXXFLAGS += -Wall -ansi -pedantic -ffast-math -std=gnu++98 -fabi-version=2 $(shell pkgconf --cflags sdl glu)
+LDFLAGS  += $(shell pkgconf --libs sdl glu) -lm
+TARGET   := 4ksdemo-by-madmartigan
+#variable de nettoyage
+RM_F := rm -f
+#variables d'instalation
+INSTALL = install
+INSTALL_DIR     = $(INSTALL) -p -d -o root -g root  -m  755
+INSTALL_PROGRAM = $(INSTALL) -p    -o root -g root  -m  755
+prefix          = /usr
+exec_prefix     = $(prefix)
+bindir          = $(prefix)/bin
+
+SOURCE          = main.cpp
+
+export
+
+all: $(TARGET)
+
+$(TARGET):
+	 $(CXX) -o $(TARGET) $(SOURCE) $(CXXFLAGS) $(LDFLAGS)
+
+install: $(TARGET)
+	$(INSTALL_DIR) $(DESTDIR)$(bindir)
+	-@$(RM_F) $(DESTDIR)$(bindir)/$(TARGET)
+	$(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)$(bindir)
+
+clean:
+	-$(RM_F) $(TARGET)
--- /dev/null
+++ b/g.h
@@ -0,0 +1,5 @@
+#include "SDL.h"
+#include "SDL_opengl.h"
+#include <math.h>
+#include <stdlib.h>
+
