--- a/mainwindow.ui
+++ b/mainwindow.ui
@@ -6001,7 +6001,7 @@
             <item>
              <widget class="QCheckBox" name="cb_FlashbackPiano">
               <property name="text">
-               <string>Played piano durring flashback</string>
+               <string>Played piano during flashback</string>
               </property>
              </widget>
             </item>
@@ -6055,7 +6055,7 @@
                      <item>
                       <widget class="QCheckBox" name="cb_tut_sub">
                        <property name="text">
-                        <string>Controling the Sub</string>
+                        <string>Controlling the Sub</string>
                        </property>
                        <property name="tristate">
                         <bool>false</bool>
--- a/lang/bchoco_de.ts
+++ b/lang/bchoco_de.ts
@@ -52,7 +52,7 @@
         <translation type="unfinished">Verschiedene Code-Hilfe / Debug; Windows QA</translation>
     </message>
     <message>
-        <source>Psx Icon Class &amp; PC string convertor for japanese</source>
+        <source>Psx Icon Class &amp; PC string converter for japanese</source>
         <translation>PSX-Symbol Klasse &amp; PC string Konvertierer für Japan</translation>
     </message>
     <message>
@@ -7359,7 +7359,7 @@
         <translation>Tutorials Angesehen</translation>
     </message>
     <message>
-        <source>Controling the Sub</source>
+        <source>Controlling the Sub</source>
         <translation>Kontrolliere das U-Boot</translation>
     </message>
     <message>
@@ -8326,7 +8326,7 @@
         <translation type="unfinished">Snowboard Minispiel</translation>
     </message>
     <message>
-        <source>Played piano durring flashback</source>
+        <source>Played piano during flashback</source>
         <translation type="unfinished">Spielte Klavier während des Flashbacks</translation>
     </message>
     <message numerus="yes">
--- a/lang/bchoco_en.ts
+++ b/lang/bchoco_en.ts
@@ -52,8 +52,8 @@
         <translation>Various code help/debug; Windows QA</translation>
     </message>
     <message>
-        <source>Psx Icon Class &amp; PC string convertor for japanese</source>
-        <translation>Psx Icon Class &amp; PC string convertor for japanese</translation>
+        <source>Psx Icon Class &amp; PC string converter for japanese</source>
+        <translation>Psx Icon Class &amp; PC string converter for japanese</translation>
     </message>
     <message>
         <source>Bug Testers</source>
@@ -7355,8 +7355,8 @@
         <translation>Tutorials Seen</translation>
     </message>
     <message>
-        <source>Controling the Sub</source>
-        <translation>Controling the Sub</translation>
+        <source>Controlling the Sub</source>
+        <translation>Controlling the Sub</translation>
     </message>
     <message>
         <source>Saving on the World Map</source>
@@ -8321,8 +8321,8 @@
         <translation>Snowboard Mini Game </translation>
     </message>
     <message>
-        <source>Played piano durring flashback</source>
-        <translation>Played piano durring flashback</translation>
+        <source>Played piano during flashback</source>
+        <translation>Played piano during flashback</translation>
     </message>
     <message numerus="yes">
         <source>Game Uses %n Save Block(s)</source>
--- a/lang/bchoco_es.ts
+++ b/lang/bchoco_es.ts
@@ -52,7 +52,7 @@
         <translation>Varios codigos. Ayuda / Debugger; Windows QA</translation>
     </message>
     <message>
-        <source>Psx Icon Class &amp; PC string convertor for japanese</source>
+        <source>Psx Icon Class &amp; PC string converter for japanese</source>
         <translation>Clase de Icono PSX y Conversor de texto de japonés para PC</translation>
     </message>
     <message>
@@ -7359,7 +7359,7 @@
         <translation>Tutoriales Vistos</translation>
     </message>
     <message>
-        <source>Controling the Sub</source>
+        <source>Controlling the Sub</source>
         <translation>Controlar el Submarino</translation>
     </message>
     <message>
@@ -8318,7 +8318,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>Played piano durring flashback</source>
+        <source>Played piano during flashback</source>
         <translation type="unfinished"></translation>
     </message>
     <message numerus="yes">
--- a/lang/bchoco_fr.ts
+++ b/lang/bchoco_fr.ts
@@ -52,7 +52,7 @@
         <translation>De l&apos;aide sur le code et du débogage varié; AQ Windows</translation>
     </message>
     <message>
-        <source>Psx Icon Class &amp; PC string convertor for japanese</source>
+        <source>Psx Icon Class &amp; PC string converter for japanese</source>
         <translation>Classe d&apos;icône PSX et convertisseur de japonais pour PC</translation>
     </message>
     <message>
@@ -7359,7 +7359,7 @@
         <translation>Tutoriels vus</translation>
     </message>
     <message>
-        <source>Controling the Sub</source>
+        <source>Controlling the Sub</source>
         <translation>Contrôle du sous-marin</translation>
     </message>
     <message>
@@ -8321,7 +8321,7 @@
         <translation type="unfinished">Mini jeu de snowboard</translation>
     </message>
     <message>
-        <source>Played piano durring flashback</source>
+        <source>Played piano during flashback</source>
         <translation type="unfinished">Joué du piano pendant le flashback</translation>
     </message>
     <message numerus="yes">
--- a/lang/bchoco_ja.ts
+++ b/lang/bchoco_ja.ts
@@ -48,7 +48,7 @@
         <translation>各種コードヘルプ/デバッグ。 Windows QA</translation>
     </message>
     <message>
-        <source>Psx Icon Class &amp; PC string convertor for japanese</source>
+        <source>Psx Icon Class &amp; PC string converter for japanese</source>
         <translation>PSX アイコン クラス &amp; 日本語文字列変換機能</translation>
     </message>
     <message>
@@ -7355,7 +7355,7 @@
         <translation>チュートリアルの表示</translation>
     </message>
     <message>
-        <source>Controling the Sub</source>
+        <source>Controlling the Sub</source>
         <translation>潜水艦の操作方法</translation>
     </message>
     <message>
@@ -8316,7 +8316,7 @@
         <translation type="unfinished">スノーボードミニゲーム</translation>
     </message>
     <message>
-        <source>Played piano durring flashback</source>
+        <source>Played piano during flashback</source>
         <translation type="unfinished">フラッシュバック中にピアノを弾いた</translation>
     </message>
     <message numerus="yes">
--- a/lang/bchoco_re.ts
+++ b/lang/bchoco_re.ts
@@ -149,8 +149,8 @@
     </message>
     <message>
         <location filename="../about.ui" line="423"/>
-        <source>Psx Icon Class &amp; PC string convertor for japanese</source>
-        <translation>Psx Icon Class &amp; PC string convertor for japanese</translation>
+        <source>Psx Icon Class &amp; PC string converter for japanese</source>
+        <translation>Psx Icon Class &amp; PC string converter for japanese</translation>
     </message>
     <message>
         <location filename="../about.ui" line="459"/>
@@ -10707,8 +10707,8 @@
     </message>
     <message>
         <location filename="../mainwindow.ui" line="6004"/>
-        <source>Played piano durring flashback</source>
-        <translation>Played piano durring flashback</translation>
+        <source>Played piano during flashback</source>
+        <translation>Played piano during flashback</translation>
     </message>
     <message>
         <location filename="../mainwindow.ui" line="6011"/>
@@ -10724,8 +10724,8 @@
     </message>
     <message>
         <location filename="../mainwindow.ui" line="6058"/>
-        <source>Controling the Sub</source>
-        <translation>Controling the Sub</translation>
+        <source>Controlling the Sub</source>
+        <translation>Controlling the Sub</translation>
     </message>
     <message>
         <location filename="../mainwindow.ui" line="6091"/>
--- a/about.ui
+++ b/about.ui
@@ -420,7 +420,7 @@
                <item>
                 <widget class="QLabel" name="label_37">
                  <property name="text">
-                  <string>Psx Icon Class &amp; PC string convertor for japanese</string>
+                  <string>Psx Icon Class &amp; PC string converter for japanese</string>
                  </property>
                 </widget>
                </item>
