--- a/view.cpp
+++ b/view.cpp
@@ -123,9 +123,9 @@
 Texture planeet,stel;
 
 Planet() :
-	planeet("planeet.png"),
-	stel("stel.png")
-	// : gras("bkglayer.png")
+	planeet("/usr/share/mleinandsharkywedtro-by-limpninja/planeet.png"),
+    stel("/usr/share/mleinandsharkywedtro-by-limpninja/stel.png")
+    // : gras("/usr/share/mleinandsharkywedtro-by-limpninja/bkglayer.png")
 {
 	for(int x=0; x<3; x++)
 		plants.push_back( new SimpelPlant() );
@@ -407,7 +407,7 @@
 		capturefake = 1;
 		printf("enabling capture support!\n");
 	}
-	ssystem.load("DISCODEL.XM");
+	ssystem.load("/usr/share/mleinandsharkywedtro-by-limpninja/DISCODEL.XM");
 	smoothCamera = Vec2d(0.0,-2.0);
 	glutInit( &argc, argv );
 	glutInitDisplayMode(GLUT_RGB | GLUT_DOUBLE | GLUT_DEPTH | GLUT_MULTISAMPLE);
@@ -415,7 +415,7 @@
 	glutCreateWindow("view");
 	capture = new Capture( 720, 576 );
 
-	frend = new FontRenderer( "fonts/Blomster.ttf" );
+	frend = new FontRenderer( "/usr/share/fonts/truetype/blomster/blomster.ttf" );
 	toverlays = new TextOverlayManager( );
 
 	//  Vec3d textColor( 0.8 +0.15,0.3+0.15,0.6+0.15 );
--- a/backdrop.cpp
+++ b/backdrop.cpp
@@ -2,8 +2,8 @@
 #include "util.h"
 #include "texture.h"
 
-Texture bluetile("paper.jpg");
-Texture gras("bkglayer.png");
+Texture bluetile("/usr/share/mleinandsharkywedtro-by-limpninja/paper.jpg");
+Texture gras("/usr/share/mleinandsharkywedtro-by-limpninja/bkglayer.png");
 
 using namespace std;
 
--- a/backdrop.h
+++ b/backdrop.h
@@ -59,7 +59,7 @@
 Texture cloudtexture;
 std::vector<Cloud> clouds;
 int maxcloudcount;
-Backdrop() : cloudtexture("cloud.png")
+Backdrop() : cloudtexture("/usr/share/mleinandsharkywedtro-by-limpninja/cloud.png")
 {
 	bkgFactor = 0.0;
 	stars = 0xffffffff;
--- a/plant.h
+++ b/plant.h
@@ -450,7 +450,7 @@
 Texture tex;
 
 public:
-SimpelPlant() : SimpelPlantSegment(), tex("stemflower_debug.png")
+SimpelPlant() : SimpelPlantSegment(), tex("/usr/share/mleinandsharkywedtro-by-limpninja/stemflower_debug.png")
 {
 	setColor( 1,1,1 );
 	segmentwidth = 0.3;
@@ -763,7 +763,7 @@
 Texture tex;
 
 public:
-SlingerPlant() : SlingerPlantSegment(), tex("slingerplant.png")
+SlingerPlant() : SlingerPlantSegment(), tex("/usr/share/mleinandsharkywedtro-by-limpninja/slingerplant.png")
 {
 	setColor( 1,1,1 );
 	segmentwidth = 0.3;
--- a/textoverlay.h
+++ b/textoverlay.h
@@ -63,7 +63,7 @@
 public:
 
 TextOverlayManager() :
-	frenderer( "fonts/Blomster.ttf" )
+	frenderer( "/usr/share/fonts/truetype/blomster/blomster.ttf" )
 {
 };
 
