--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,33 @@
-CFLAGS = `sdl-config --cflags`
-LDLIBS = `sdl-config --libs`
+#variables de compilation
+SHELL = /bin/sh
+CC    = gcc
+CFLAGS   +=  -std=c99 -Wall -ffast-math -pipe -fstrict-aliasing $(shell pkgconf --cflags sdl glu)
+LDFLAGS  += $(shell pkgconf --libs sdl glu) -lm
+TARGET   = analeptique-by-acetate
+#variables d'instalation
+INSTALL = install
+INSTALL_DIR     = $(INSTALL) -p -d -o root -g root  -m  755
+INSTALL_PROGRAM = $(INSTALL) -p    -o root -g root  -m  755
+prefix          = /usr
+exec_prefix     = $(prefix)
+bindir          = $(prefix)/bin
+#variable de nettoyage
+RM_F := rm -f
 
-all:
-	gcc $(CFLAGS) $(LDLIBS) analeptique.c -o analeptique -ansi -O2
-	@echo Analeptique \~ source size \(bytes\) : && cat analeptique.c | perl -pe 's/[\s\r\n]//g' | wc -c
\ No newline at end of file
+
+SOURCE          = analeptique.c
+
+export
+
+all: $(TARGET)
+
+$(TARGET):
+	 $(CC) -o $(TARGET) $(SOURCE) $(CFLAGS) $(LDFLAGS)
+
+install: $(TARGET)
+	$(INSTALL_DIR) $(DESTDIR)$(bindir)
+	-@$(RM_F) $(DESTDIR)$(bindir)/$(TARGET)
+	$(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)$(bindir)
+
+clean:
+	-$(RM_F) $(TARGET)
--- /dev/null
+++ b/s.h
@@ -0,0 +1,4 @@
+#include "SDL.h"
+#include <math.h>
+#include <stdlib.h>
+
