--- a/main.c
+++ b/main.c
@@ -91,6 +91,13 @@
 #endif
 }
 
+void ToggleFullscreen(SDL_Window* Window) {
+	Uint32 FullscreenFlag = SDL_WINDOW_FULLSCREEN;
+	int IsFullscreen = SDL_GetWindowFlags(Window) & FullscreenFlag;
+	SDL_SetWindowFullscreen(Window, IsFullscreen ? 0 : FullscreenFlag);
+	SDL_ShowCursor(IsFullscreen);
+}
+
 int main(int argc,char *argv[])
 {
 	int quit=0,n,i,*pal,last=0,pos=0,tmp,r,g,b,*pix,origtime=-1;
@@ -206,6 +213,7 @@
 				switch (e.key.keysym.sym) {
 					case SDLK_ESCAPE: quit=1; break;
 					case SDLK_q: quit=1; break;
+					case SDLK_f: ToggleFullscreen(sdlWindow); break;
 				}
 			 }
 			if (e.window.windowID == windowID)  {
