--- a/console_source/console_source.pro
+++ b/console_source/console_source.pro
@@ -3,6 +3,11 @@
 
 XCONFIG += use_dex
 
+QMAKE_CPPFLAGS *= $(shell dpkg-buildflags --get CPPFLAGS)
+QMAKE_CFLAGS   *= $(shell dpkg-buildflags --get CFLAGS)
+QMAKE_CXXFLAGS *= $(shell dpkg-buildflags --get CXXFLAGS)
+QMAKE_LFLAGS   *= $(shell dpkg-buildflags --get LDFLAGS)
+
 CONFIG += c++11
 
 TARGET = diec
--- a/gui_source/gui_source.pro
+++ b/gui_source/gui_source.pro
@@ -53,5 +53,10 @@
     ICON = ../icons/main.icns
 }
 
+QMAKE_CPPFLAGS *= $(shell dpkg-buildflags --get CPPFLAGS)
+QMAKE_CFLAGS   *= $(shell dpkg-buildflags --get CFLAGS)
+QMAKE_CXXFLAGS *= $(shell dpkg-buildflags --get CXXFLAGS)
+QMAKE_LFLAGS   *= $(shell dpkg-buildflags --get LDFLAGS)
+
 DISTFILES += \
     CMakeLists.txt
--- a/die_source.pro
+++ b/die_source.pro
@@ -5,5 +5,4 @@
 SUBDIRS         += lite_source
 
 target.path = $$PWD
-target.commands = bash -x $$PWD/install.sh
 INSTALLS += target
