--- a/build_lin64.sh
+++ b/build_lin64.sh
@@ -1,5 +1,5 @@
 #!/bin/bash -x
-QT_PATH=$HOME/Qt/5.12.8/gcc_64
+#QT_PATH=$HOME/Qt/5.12.8/gcc_64
 RELEASE_VERSION=$(cat "release_version.txt")
 echo $RELEASE_VERSION
 SOURCE_PATH=$PWD
@@ -19,7 +19,7 @@
     rm -rf Makefile
     rm -rf Makefile.Release
     rm -rf Makefile.Debug
-    rm -rf object_script.*     
+    rm -rf object_script.*
 
     cd $SOURCE_PATH
 }
--- a/console_source/console_source.pro
+++ b/console_source/console_source.pro
@@ -5,6 +5,11 @@
 
 XCONFIG += use_dex
 
+QMAKE_CPPFLAGS *= $(shell dpkg-buildflags --get CPPFLAGS)
+QMAKE_CFLAGS   *= $(shell dpkg-buildflags --get CFLAGS)
+QMAKE_CXXFLAGS *= $(shell dpkg-buildflags --get CXXFLAGS)
+QMAKE_LFLAGS   *= $(shell dpkg-buildflags --get LDFLAGS)
+
 CONFIG += c++11
 
 TARGET = diec
--- a/gui_source/gui_source.pro
+++ b/gui_source/gui_source.pro
@@ -48,3 +48,8 @@
 macx {
     ICON = ../icons/main.icns
 }
+
+QMAKE_CPPFLAGS *= $(shell dpkg-buildflags --get CPPFLAGS)
+QMAKE_CFLAGS   *= $(shell dpkg-buildflags --get CFLAGS)
+QMAKE_CXXFLAGS *= $(shell dpkg-buildflags --get CXXFLAGS)
+QMAKE_LFLAGS   *= $(shell dpkg-buildflags --get LDFLAGS)
