--- a/XArchive/3rdparty/bzip2/bzip2.pri
+++ b/XArchive/3rdparty/bzip2/bzip2.pri
@@ -17,12 +17,7 @@
 }
 unix:!macx {
     BITSIZE = $$system(getconf LONG_BIT)
-    if (contains(BITSIZE, 64)) {
-        LIBS +=  $$PWD/libs/lin64/libbzip2.a
-    }
-    if (contains(BITSIZE, 32)) {
-        LIBS +=  $$PWD/libs/lin32/libbzip2.a
-    }
+    LIBS += -lbz2
 }
 unix:macx {
     LIBS +=  $$PWD/libs/mac/libbzip2.a
--- a/XArchive/xarchive.pri
+++ b/XArchive/xarchive.pri
@@ -22,12 +22,12 @@
 
 !contains(XCONFIG, zlib) {
     XCONFIG += zlib
-    include($$PWD/3rdparty/zlib/zlib.pri)
+    LIBS += -lz
 }
 
 !contains(XCONFIG, bzip2) {
     XCONFIG += bzip2
-    include($$PWD/3rdparty/bzip2/bzip2.pri)
+    LIBS += -lbz2
 }
 
 !contains(XCONFIG, lzma) {
--- a/XQwt/xqwt.pri
+++ b/XQwt/xqwt.pri
@@ -3,5 +3,6 @@
 
 !contains(XCONFIG, qwt) {
     XCONFIG += qwt
-    include($$PWD/3rdparty/qwt/qwt.pri)
+    INCLUDEPATH += /usr/include/qwt/
+	LIBS += -lqwt-qt5
 }
