--- a/XOptions/xoptions.cpp
+++ b/XOptions/xoptions.cpp
@@ -232,7 +232,7 @@
     if (bIsNative) {
         pSettings = new QSettings;
     } else {
-        pSettings = new QSettings(getApplicationDataPath() + QDir::separator() + QString("%1").arg(g_sName), QSettings::IniFormat);
+        pSettings = new QSettings(getConfigPath()+QDir::separator()+QString("%1").arg(g_sName),QSettings::IniFormat);
     }
 
 #ifdef QT_DEBUG
@@ -622,6 +622,14 @@
     return sResult;
 }
 
+QString XOptions::getConfigPath()
+{
+    QString sResult;
+    sResult=QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
+
+    return sResult;
+}
+
 void XOptions::setLastDirectory(QString sPathName)
 {
     if (getValue(ID_FILE_SAVELASTDIRECTORY).toBool()) {
@@ -881,7 +889,7 @@
         pComboBox->addItem("English", "");
         pComboBox->addItem("System", "System");
 
-        QList<QString> listFileNames = getAllFilesFromDirectory(getApplicationLangPath(), "*.qm");
+        QList<QString> listFileNames = getAllFilesFromDirectory("/usr/share/detectiteasy/lang","*.qm");
 
         qint32 nNumberOfRecords = listFileNames.count();
 
@@ -908,7 +916,7 @@
     } else if (id == ID_VIEW_QSS) {
         pComboBox->addItem("Default", "");
 
-        QList<QString> listFileNames = getAllFilesFromDirectory(getApplicationQssPath(), "*.qss");
+        QList<QString> listFileNames = getAllFilesFromDirectory("/usr/share/detectiteasy/qss","*.qss");
 
         qint32 nNumberOfRecords = listFileNames.count();
 
@@ -1612,7 +1620,7 @@
                 sResult = sApplicationDirPath.section("/", 0, 2);
             }
 
-            sResult += QString("/usr/lib/%1").arg(qApp->applicationName());
+            sResult += QString("/usr/share/detectiteasy").arg(qApp->applicationName());
         }
     } else {
         sResult = sApplicationDirPath;
--- a/XOptions/xoptions.h
+++ b/XOptions/xoptions.h
@@ -26,6 +26,7 @@
 #include <QSettings>
 #include <QSysInfo>
 #include <QTranslator>
+#include <QStandardPaths>
 //#include <QVector>
 #ifdef QT_GUI_LIB
 #include <QApplication>
@@ -243,6 +244,7 @@
     static QList<QString> getAllFilesFromDirectory(QString sDirectory, QString sExtension);
     static bool checkNative(QString sIniFileName);
     QString getApplicationDataPath();
+    static QString getConfigPath();
     static QString getTitle(QString sName, QString sVersion, bool bShowOS = true);
 #ifdef QT_GUI_LIB
     void setCheckBox(QCheckBox *pCheckBox, ID id);
