--- a/XOptions/xoptions.cpp
+++ b/XOptions/xoptions.cpp
@@ -28,7 +28,7 @@
 void XOptions::setName(QString sName)
 {
     this->g_sName=sName;
-    this->g_sFilePath=getApplicationDataPath()+QDir::separator()+QString("%1").arg(sName);
+    this->g_sFilePath=getConfigPath()+QDir::separator()+QString("%1").arg(sName);
 }
 
 void XOptions::setValueIDs(QList<ID> listVariantIDs)
@@ -73,8 +73,8 @@
                 case ID_STYLE:                  varDefault="Fusion";        break;
                 case ID_LANG:                   varDefault="System";        break;
                 case ID_QSS:                    varDefault="veles";         break;
-                case ID_DBPATH:                 varDefault="$data/db";      break;
-                case ID_INFOPATH:               varDefault="$data/info";    break;
+                case ID_DBPATH:                 varDefault="/usr/share/detectiteasy/db";      break;
+                case ID_INFOPATH:               varDefault="/usr/share/detectiteasy/info";    break;
                 case ID_SCANENGINE:             varDefault="die";           break;
                 case ID_ROOTPATH:               varDefault="";              break;
                 case ID_DATAPATH:               varDefault="$data/data";    break;
@@ -183,6 +183,15 @@
     return sResult;
 }
 
+QString XOptions::getConfigPath()
+{
+    QString sResult;
+    sResult=QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
+
+    return sResult;
+}
+
+
 void XOptions::setLastDirectory(QString sValue)
 {
     QFileInfo fi(sValue);
@@ -297,7 +306,7 @@
         pComboBox->addItem("English","");
         pComboBox->addItem("System","System");
 
-        QList<QString> listFileNames=getAllFilesFromDirectory(getApplicationLangPath(),"*.qm");
+        QList<QString> listFileNames=getAllFilesFromDirectory("/usr/share/detectiteasy/lang","*.qm");
 
         int nNumberOfRecords=listFileNames.count();
 
@@ -322,7 +331,7 @@
     {
         pComboBox->addItem("Default","");
 
-        QList<QString> listFileNames=getAllFilesFromDirectory(getApplicationQssPath(),"*.qss");
+        QList<QString> listFileNames=getAllFilesFromDirectory("/usr/share/detectiteasy/qss","*.qss");
 
         int nNumberOfRecords=listFileNames.count();
 
@@ -515,7 +524,7 @@
 #ifdef Q_OS_MAC
     sResult=qApp->applicationDirPath()+"/../Resources";
 #else
-    sResult=qApp->applicationDirPath();
+    sResult="/usr/share/detectiteasy";
 #endif
     return sResult;
 }
--- a/XOptions/xoptions.h
+++ b/XOptions/xoptions.h
@@ -26,6 +26,7 @@
 #include <QDir>
 #include <QSettings>
 #include <QTranslator>
+#include <QStandardPaths>
 #ifdef QT_GUI_LIB
 #include <QWidget>
 #include <QCheckBox>
@@ -94,6 +95,7 @@
     static QString getApplicationQssPath();
     static QList<QString> getAllFilesFromDirectory(QString sDirectory,QString sExtension);
     static QString getApplicationDataPath();
+    static QString getConfigPath();
 #ifdef QT_GUI_LIB
     void setCheckBox(QCheckBox *pCheckBox,ID id);
     void getCheckBox(QCheckBox *pCheckBox,ID id);
