--- a/XOptions/xoptions.cpp
+++ b/XOptions/xoptions.cpp
@@ -97,7 +97,7 @@
     }
     else
     {
-        pSettings=new QSettings(getApplicationDataPath()+QDir::separator()+QString("%1").arg(g_sName),QSettings::IniFormat);
+        pSettings=new QSettings(getConfigPath()+QDir::separator()+QString("%1").arg(g_sName),QSettings::IniFormat);
     }
 
     int nNumberOfIDs=g_listValueIDs.count();
@@ -128,8 +128,8 @@
                 case ID_STYLE:                  varDefault="Fusion";                break;
                 case ID_LANG:                   varDefault="System";                break;
                 case ID_QSS:                    varDefault="veles";                 break;
-                case ID_DATABASEPATH:           varDefault="$data/db";              break;
-                case ID_INFOPATH:               varDefault="$data/info";            break;
+                case ID_DATABASEPATH:           varDefault="/usr/share/detectiteasy/db";      break;
+                case ID_INFOPATH:               varDefault="/usr/share/detectiteasy/info";    break;
                 case ID_SCANENGINE:             varDefault="die";                   break;
                 case ID_ROOTPATH:               varDefault="";                      break;
                 case ID_DATAPATH:               varDefault="$data/data";            break;
@@ -286,6 +286,15 @@
     return sResult;
 }
 
+QString XOptions::getConfigPath()
+{
+    QString sResult;
+    sResult=QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
+
+    return sResult;
+}
+
+
 void XOptions::setLastDirectory(QString sValue)
 {
     QFileInfo fi(sValue);
@@ -419,7 +428,7 @@
         pComboBox->addItem("English","");
         pComboBox->addItem("System","System");
 
-        QList<QString> listFileNames=getAllFilesFromDirectory(getApplicationLangPath(),"*.qm");
+        QList<QString> listFileNames=getAllFilesFromDirectory("/usr/share/detectiteasy/lang","*.qm");
 
         int nNumberOfRecords=listFileNames.count();
 
@@ -444,7 +453,7 @@
     {
         pComboBox->addItem("Default","");
 
-        QList<QString> listFileNames=getAllFilesFromDirectory(getApplicationQssPath(),"*.qss");
+        QList<QString> listFileNames=getAllFilesFromDirectory("/usr/share/detectiteasy/qss","*.qss");
 
         int nNumberOfRecords=listFileNames.count();
 
@@ -688,7 +697,7 @@
             sResult=sApplicationDirPath.section("/",0,2);
         }
 
-        sResult+=QString("/usr/lib/%1").arg(qApp->applicationName());
+        sResult="/usr/share/detectiteasy";
     }
     else
     {
--- a/XOptions/xoptions.h
+++ b/XOptions/xoptions.h
@@ -25,6 +25,7 @@
 #include <QDir>
 #include <QSettings>
 #include <QTranslator>
+#include <QStandardPaths>
 #ifdef QT_GUI_LIB
 #include <QWidget>
 #include <QCheckBox>
@@ -101,6 +102,7 @@
     static QList<QString> getAllFilesFromDirectory(QString sDirectory,QString sExtension);
     static bool checkNative();
     QString getApplicationDataPath();
+    static QString getConfigPath();
 #ifdef QT_GUI_LIB
     void setCheckBox(QCheckBox *pCheckBox,ID id);
     void getCheckBox(QCheckBox *pCheckBox,ID id);
