--- a/XOptions/xoptions.cpp
+++ b/XOptions/xoptions.cpp
@@ -115,7 +115,7 @@
     }
     else
     {
-        pSettings=new QSettings(getApplicationDataPath()+QDir::separator()+QString("%1").arg(g_sName),QSettings::IniFormat);
+        pSettings=new QSettings(getConfigPath()+QDir::separator()+QString("%1").arg(g_sName),QSettings::IniFormat);
     }
 
 #ifdef QT_DEBUG
@@ -153,8 +153,8 @@
                 case ID_STYLE:                  varDefault="Fusion";                break; // TODO Check styles in OSX and Linux
                 case ID_LANG:                   varDefault="System";                break;
                 case ID_QSS:                    varDefault="";                      break;
-                case ID_DATABASEPATH:           varDefault="$data/db";              break;
-                case ID_INFOPATH:               varDefault="$data/info";            break;
+                case ID_DATABASEPATH:           varDefault="/usr/share/detectiteasy/db";      break;
+                case ID_INFOPATH:               varDefault="/usr/share/detectiteasy/info";    break;
                 case ID_SCANENGINE:             varDefault="die";                   break;
                 case ID_DISASMSYNTAX:           varDefault="";                      break;
                 case ID_ROOTPATH:               varDefault="";                      break;
@@ -328,6 +328,14 @@
     return sResult;
 }
 
+QString XOptions::getConfigPath()
+{
+    QString sResult;
+    sResult=QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
+
+    return sResult;
+}
+
 void XOptions::setLastDirectory(QString sPathName)
 {
     QFileInfo fi(sPathName);
@@ -532,7 +540,7 @@
         pComboBox->addItem("English","");
         pComboBox->addItem("System","System");
 
-        QList<QString> listFileNames=getAllFilesFromDirectory(getApplicationLangPath(),"*.qm");
+        QList<QString> listFileNames=getAllFilesFromDirectory("/usr/share/detectiteasy/lang","*.qm");
 
         qint32 nNumberOfRecords=listFileNames.count();
 
@@ -557,7 +565,7 @@
     {
         pComboBox->addItem("Default","");
 
-        QList<QString> listFileNames=getAllFilesFromDirectory(getApplicationQssPath(),"*.qss");
+        QList<QString> listFileNames=getAllFilesFromDirectory("/usr/share/detectiteasy/qss","*.qss");
 
         qint32 nNumberOfRecords=listFileNames.count();
 
@@ -1010,7 +1018,7 @@
             sResult=sApplicationDirPath.section("/",0,2);
         }
 
-        sResult+=QString("/usr/lib/%1").arg(qApp->applicationName());
+        sResult="/usr/share/detectiteasy";
     }
     else
     {
--- a/XOptions/xoptions.h
+++ b/XOptions/xoptions.h
@@ -26,6 +26,7 @@
 #include <QSettings>
 #include <QSysInfo>
 #include <QTranslator>
+#include <QStandardPaths>
 #ifdef QT_GUI_LIB
 #include <QApplication>
 #include <QCheckBox>
@@ -131,6 +132,7 @@
     static QList<QString> getAllFilesFromDirectory(QString sDirectory,QString sExtension);
     static bool checkNative();
     QString getApplicationDataPath();
+    static QString getConfigPath();
     static QString getTitle(QString sName,QString sVersion);
 #ifdef QT_GUI_LIB
     void setCheckBox(QCheckBox *pCheckBox,ID id);
