--- a/main.c
+++ b/main.c
@@ -101,6 +101,9 @@
 #define DEFAULT_HEIGHT 480
 #endif
 
+#define DEMO_XASPECT 4.0f
+#define DEMO_YASPECT 3.0f
+
 #ifdef __NO_MOUSE__
 SDL_Joystick *joy;
 int joyx=0,joyy=0,joyb1=0,joyb2=0;
@@ -111,6 +114,27 @@
 
 int clicked, clickx, clicky;
 
+void resize_window(SDL_Renderer *sdlRenderer, int w, int h) {
+    SDL_Rect newWindowSize;
+    newWindowSize.x = 0;
+    newWindowSize.y = 0;
+
+    float aspect = (float)w / (float)h;
+    if (aspect > DEMO_XASPECT / DEMO_YASPECT) {
+        newWindowSize.w = (int)((float)h * DEMO_XASPECT / DEMO_YASPECT);
+        newWindowSize.x = (w - newWindowSize.w) / 2;
+        newWindowSize.h = h;
+    } else {
+        newWindowSize.h = (int)((float)w * DEMO_YASPECT / DEMO_XASPECT);
+        newWindowSize.y = (h - newWindowSize.h) / 2;
+        newWindowSize.w = w;
+    }
+
+    SDL_Rect viewPort;
+    SDL_RenderGetViewport(sdlRenderer, &viewPort);
+    SDL_RenderSetViewport(sdlRenderer, &newWindowSize);
+}
+
 void do_the_music( void *dummy, int8 *stream, int length )
 {
   int16 *out;
@@ -487,12 +511,12 @@
 #endif
   
 #ifdef __SW_SURFACE__
-  flags = SDL_WINDOW_FULLSCREEN;
+  flags = SDL_WINDOW_RESIZABLE | SDL_WINDOW_FULLSCREEN;
 #else
   #ifdef __HW_ONLY_WINDOWED__
-    flags = SDL_WINDOW_FULLSCREEN;
+    flags = SDL_WINDOW_RESIZABLE | SDL_WINDOW_FULLSCREEN;
   #else
-    flags = SDL_WINDOW_FULLSCREEN;
+    flags = SDL_WINDOW_RESIZABLE | SDL_WINDOW_FULLSCREEN;
   #endif
 #endif
 
@@ -543,6 +567,8 @@
 
   sdlWindow = SDL_CreateWindow("Planet Hively",SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED,sw,sh,flags);
   sdlRenderer = SDL_CreateRenderer(sdlWindow, -1, SDL_RENDERER_PRESENTVSYNC);
+  SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "1");
+  Uint32 windowID = SDL_GetWindowID(sdlWindow);
   screen = SDL_CreateRGBSurfaceWithFormat(0, sw, sh, BITSPERPIXEL,SDL_PIXELFORMAT_RGB565); //video buffer
   sdlTexture = SDL_CreateTextureFromSurface(sdlRenderer,screen);// texture from video buffer
   if( !screen )
@@ -779,6 +805,14 @@
   #endif // SPECIAL JOYSTICK TYPES
 #endif
 
+		case SDL_WINDOWEVENT:
+          if (event.window.windowID == windowID)  {
+		  if (event.window.event == SDL_WINDOWEVENT_RESIZED)  {
+			   resize_window(sdlRenderer,event.window.data1,event.window.data2);
+			  }
+		  }
+          break;
+
         case SDL_MOUSEMOTION:
           dragx = event.motion.x;
           dragy = event.motion.y;
