From: Stefan Fritsch <sf@sfritsch.de>
Date: Fri, 24 Jul 2009 15:23:32 +0200
Subject: don't return 0 on error

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=433757
---
 mp3gain/mp3gain.c | 28 ++++++++++++++++------------
 1 file changed, 16 insertions(+), 12 deletions(-)

--- a/mp3gain/mp3gain.c
+++ b/mp3gain/mp3gain.c
@@ -2012,6 +2012,7 @@
 					} else {
 						fprintf(stderr,"No undo information in %s\n",argv[mainloop]);
 					}
+					gSuccess = 0;
 				}
 		  }
 	  }
@@ -2082,8 +2083,9 @@
 #else
 		  if ((inf == NULL)&&(tagInfo[mainloop].recalc > 0)) {
 #endif
-			  fprintf(stdout, "Can't open %s for reading\n",argv[mainloop]);
-              fflush(stdout);
+			  fprintf(stderr, "Can't open %s for reading\n",argv[mainloop]);
+              fflush(stderr);
+              gSuccess = 0;
 		  }
 		  else {
 #ifdef AACGAIN
@@ -2168,8 +2170,9 @@
 				if (!ok) {
 #endif
                     if (!BadLayer) {
-						fprintf(stdout,"Can't find any valid MP3 frames in file %s\n",argv[mainloop]);
-                        fflush(stdout);
+						fprintf(stderr,"Can't find any valid MP3 frames in file %s\n",argv[mainloop]);
+                        fflush(stderr);
+                        gSuccess = 0;
                     }
 				}
 				else {
@@ -2204,8 +2207,8 @@
 								(Xingcheck[0] == 'I' && Xingcheck[1] == 'n' && Xingcheck[2] == 'f' && Xingcheck[3] == 'o')) {
 							bitridx = (curframe[2] >> 4) & 0x0F;
 							if (bitridx == 0) {
-								fprintf(stdout, "%s is free format (not currently supported)\n",curfilename);
-								fflush(stdout);
+								fprintf(stderr, "%s is free format (not currently supported)\n",curfilename);
+								fflush(stderr);
 								ok = 0;
 							}
 							else {
@@ -2248,8 +2251,8 @@
 						while (ok) {
 							bitridx = (curframe[2] >> 4) & 0x0F;
 							if (bitridx == 0) {
-								fprintf(stdout,"%s is free format (not currently supported)\n",curfilename);
-								fflush(stdout);
+								fprintf(stderr,"%s is free format (not currently supported)\n",curfilename);
+								fflush(stderr);
 								ok = 0;
 							}
 							else {
@@ -2335,8 +2338,9 @@
 					}
 
 					if (dBchange == GAIN_NOT_ENOUGH_SAMPLES) {
-						fprintf(stdout,"Not enough samples in %s to do analysis\n",argv[mainloop]);
-                        fflush(stdout);
+						fprintf(stderr,"Not enough samples in %s to do analysis\n",argv[mainloop]);
+                        fflush(stderr);
+                        gSuccess = 0;
 						numFiles--;
 					}
 					else {
@@ -2476,8 +2480,8 @@
 		}
 
 		if (dBchange == GAIN_NOT_ENOUGH_SAMPLES) {
-			fprintf(stdout,"Not enough samples in mp3 files to do analysis\n");
-            fflush(stdout);
+			fprintf(stderr,"Not enough samples in mp3 files to do analysis\n");
+            fflush(stderr);
 		}
 		else {
 			Float_t maxmaxsample;
