From: Stefan Fritsch <sf@sfritsch.de>
Date: Sat, 29 Sep 2007 11:12:27 +0200
Subject: Fix segfault when files are not writable

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=444373
---
 mp3gain/mp3gain.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

--- a/mp3gain/mp3gain.c
+++ b/mp3gain/mp3gain.c
@@ -762,6 +762,7 @@
 			inf = NULL;
             passError(MP3GAIN_UNSPECIFED_ERROR, 3,
                 "\nCan't open ", outfilename, " for temp writing\n");
+            NowWriting = 0;
 			return M3G_ERR_CANT_MAKE_TMP;
 		} 
  
@@ -772,10 +773,13 @@
   }
 
   if (inf == NULL) {
-	  if (UsingTemp && (outf != NULL))
+	  if (UsingTemp && (outf != NULL)) {
 		  fclose(outf);
+		  outf = NULL;
+	  }
 	  passError( MP3GAIN_UNSPECIFED_ERROR, 3,
           "\nCan't open ", filename, " for modifying\n");
+      NowWriting = 0;
 	  return M3G_ERR_CANT_MODIFY_FILE;
   }
   else {
@@ -968,6 +972,7 @@
 		fclose(inf);
 		if (UsingTemp) {
 			fclose(outf);
+			outf = NULL;
 			deleteFile(outfilename);
 			free(outfilename);
 			passError(MP3GAIN_CANCELLED,2,"Cancelled processing of ",filename);
@@ -977,6 +982,7 @@
 		}
 		if (saveTime) 
 		  fileTime(filename, setStoredTime);		
+		NowWriting = 0;
 		return;
 	}
 #endif
@@ -1041,12 +1047,14 @@
 		fclose(outf);
 		fclose(inf);
 		inf = NULL;
+		outf = NULL;
         
         if (outlength != inlength) {
             deleteFile(outfilename);
 			passError( MP3GAIN_UNSPECIFED_ERROR, 3,
                 "Not enough temp space on disk to modify ", filename, 
                 "\nEither free some space, or switch off \"temp file\" option with -T\n");
+            NowWriting = 0;
             return M3G_ERR_NOT_ENOUGH_TMP_SPACE;
         }
         else {
@@ -1055,6 +1063,7 @@
 				deleteFile(outfilename); //try to delete tmp file
 				passError( MP3GAIN_UNSPECIFED_ERROR, 3,
                     "Can't open ", filename, " for modifying\n");
+                NowWriting = 0;
 			    return M3G_ERR_CANT_MODIFY_FILE;
 		    }
 		    if (moveFile(outfilename, filename)) {
@@ -1063,6 +1072,7 @@
                     "\nThe mp3 was correctly modified, but you will need to re-name ", 
                     outfilename, " to ", filename, 
                     " yourself.\n");
+                   NowWriting = 0;
 			    return M3G_ERR_RENAME_TMP;
 		    };
 		    if (saveTime)
