--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 SHELL = /bin/sh
 CC    = gcc
 CFLAGS += -O3 -flto -Wall -Wextra -Wmissing-prototypes -Wformat-security -Wno-unused-parameter -Isrc
-LDFLAGS +=-Wall
+LDFLAGS +=-Wall -flto
 RM_F = rm -f
 INSTALL = install
 INSTALL_DIR     = $(INSTALL) -p -d -o root -g root  -m  755
@@ -11,6 +11,8 @@
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
 
+PLUGIN := --plugin=$(shell gcc --print-file-name=liblto_plugin.so)
+
 #parallel compilation if available
 ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
  NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
@@ -75,13 +77,13 @@
 
 DEARK1_A:=$(OBJDIR)/src/deark1.a
 $(DEARK1_A): $(OFILES_DEARK1)
-	ar rcs $@ $^
+	ar rcs $(PLUGIN) $@ $^
 
 DEARK2_A:=$(OBJDIR)/src/deark2.a
 $(DEARK2_A): $(OFILES_DEARK2)
-	ar rcs $@ $^
+	ar rcs $(PLUGIN) $@ $^
 
-ARFLAGS:=urcs
+ARFLAGS:=urcs $(PLUGIN)
 MODS_AB_A:=$(OBJDIR)/modules/mods-ab.a
 MODS_CH_A:=$(OBJDIR)/modules/mods-ch.a
 MODS_IO_A:=$(OBJDIR)/modules/mods-io.a
@@ -115,7 +117,7 @@
 	$(INSTALL_PROGRAM) $(DEARK_EXE) $(DESTDIR)$(bindir)
 
 clean:
-	rm -f $(OBJDIR)/src/*.[oad] $(OBJDIR)/modules/*.[oad] $(DEARK_EXE)
+	-@$(RM_F) -f $(OBJDIR)/src/*.[oad] $(OBJDIR)/modules/*.[oad] $(DEARK_EXE)
 
 ifeq ($(MAKECMDGOALS),dep)
 
