--- a/Makefile
+++ b/Makefile
@@ -4,6 +4,7 @@
 # -Marq 2001
 
 # Platform (LINUX/IRIX)
+SHELL = /bin/sh
 PLATFORM = LINUX
 # YES for sound, NO or whatever for no sound
 SOUND = YES
@@ -12,27 +13,41 @@
 
 # General defines
 CC = g++
-COPT = -O2 -ffast-math -fomit-frame-pointer
-DEF = 
-GL = -lglut -lGL -lGLU
-LIBS = -lm
+CXXFLAGS += -Wall -pipe $(shell pkgconf --cflags glu) -std=gnu++98 -fabi-version=2
+DEF =
+GL = -lglut $(shell pkgconf --libs glu)
+LDFLAGS += -lm
+RM_F = rm -f
+
+INSTALL = install
+INSTALL_DIR     = $(INSTALL) -p -d -o root -g root  -m  755
+INSTALL_PROGRAM = $(INSTALL) -p    -o root -g root  -m  755
+INSTALL_FILE    = $(INSTALL) -p    -o root -g root  -m  644
+
+TARGET          = neandertaler
+
+PREFIX          = /usr
+BINDIR          = $(PREFIX)/bin
+SHAREDIR        = $(PREFIX)/share
+TARGETSHAREDIR  = $(PREFIX)/share/neandertaler/data
+
+export
 
 # Linux (or XFree) specific
 ifeq ($(PLATFORM),LINUX)
-X = -L/usr/X11R6/lib -lX11 -lXext -lXmu -lXi
-LIBS += -ldl
-DEF += -DBSDENV
-endif
-
-# SGI specific
-ifeq ($(PLATFORM),IRIX)
-X = -L/usr/lib32 -lX11 -lXext -lXmu -lXi
+	X = $(shell pkgconf x11 xext xmu xi --libs)
+	LIBS += -ldl
+	DEF += -DBSDENV
+	ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+		NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+		MAKEFLAGS += -j$(NUMJOBS)
+	endif
 endif
 
 # If sound enabled
 ifeq ($(SOUND),YES)
 DEF += -DSOUND
-LIBS += -lmikmod
+LDFLAGS += -lmikmod
 endif
 
 # If old GLUT version
@@ -40,13 +55,23 @@
 DEF += -DOLDGLUT
 endif
 
-neandertaler: neandertaler_mainloop.o main.cc
-	$(CC) $(COPT) main.cc neandertaler_mainloop.o -o neandertaler \
-	$(DEF) $(GL) $(X) $(LIBS)
-	strip neandertaler
+all: $(TARGET)
+
+$(TARGET): neandertaler_mainloop.o main.cc
+	$(CC) $(CXXFLAGS) main.cc neandertaler_mainloop.o -o $(TARGET) $(DEF) $(GL) $(X) $(LDFLAGS)
 
 neandertaler_mainloop.o: neandertaler_mainloop.cc
-	$(CC) $(COPT) -c neandertaler_mainloop.cc $(DEF)
+	$(CC) $(CXXFLAGS) -c neandertaler_mainloop.cc $(DEF)
+
+clean:
+	-$(RM_F) $(TARGET) *.o
+
+install: $(TARGET)
+	$(INSTALL_DIR) $(DESTDIR)$(BINDIR)
+	-@$(RM_F) $(DESTDIR)$(BINDIR)/$(TARGET)
+	$(INSTALL_PROGRAM) $(TARGET) $(DESTDIR)$(BINDIR)
+	$(INSTALL_DIR) $(DESTDIR)$(TARGETSHAREDIR)
+	$(INSTALL_FILE) data/* $(DESTDIR)$(TARGETSHAREDIR)
 
-clean: 
-	rm neandertaler *.o *~
+uninstall:
+	-$(RM_F) $(DESTDIR)$(BINDIR)/$(TARGET)
