--- a/Makefile
+++ b/Makefile
@@ -1,20 +1,41 @@
+#variables de compilation
+SHELL := /bin/sh
 CC	:= gcc
 
-CFLAGS	:= -W -Wall -I/usr/include `sdl-config --cflags`
-CFLAGS	+= -O3 -ffast-math -fpeephole2 -fexpensive-optimizations
+CFLAGS	:= -W -Wall -I/usr/include $(shell sdl-config --cflags)
+CFLAGS	+= -O3 -flto -ffast-math -fpeephole2 -fexpensive-optimizations
 #CFLAGS	+= -g
 
+TARGET   := fracturesandrust-by-atomimaitokahvi
+#variable de nettoyage
+RM_F := rm -f
+
+#si code gcc4 compilé via gcc5 le standard par défaut était different
+GCCVERSIONGTEQ5 := $(shell expr `gcc -dumpversion | cut -f1 -d.` \>= 5)
+ifeq "$(GCCVERSIONGTEQ5)" "1"
+    CFLAGS += -std=gnu90
+    CXXFLAGS += -fabi-compat-version=2
+endif
+#compilation en paralelle si disponible
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ MAKEFLAGS += -j$(NUMJOBS)
+endif
+
+export
+
 all: main script-buildbg script-filter script-buildcr script-loop script-addtime
 
 clean:
-	rm -f *.o
-	rm -f main script-buildbg script-filter script-buildcr script-loop script-addtime
+	-@$(RM_F) *.o
+	-@$(RM_F) main script-buildbg script-filter script-buildcr script-loop script-addtime
+	-@$(RM_F) $(TARGET)
 
 main: attract.o convolve.o filter.o input.o interpolate.o im.o list.o main.o pnm.o script.o util.o xmalloc.o tuherrus.o suodatin.o
-	$(CC) -o $@ $^ `sdl-config --libs` -lvorbisfile
+	$(CC) -flto -o $(TARGET) $^ $(shell sdl-config --libs) -lvorbisfile -lm
 
 editor: convolve.o editor.o filter.o input.o interpolate.o im.o list.o pnm.o script.o util.o xmalloc.o
-	$(CC) -o $@ $^ `sdl-config --libs`
+	$(CC) -flto -o $@ $^ $(shell sdl-config --libs)
 
 script-buildbg: script-buildbg.c
 	$(CC) -W -Wall -o $@ $^ -lm
