--- a/Makefile
+++ b/Makefile
@@ -1,38 +1,53 @@
+#variables de compilation
+SHELL := /bin/sh
 CC	:= gcc
 
-CFLAGS	:= -W -Wall -I/usr/include `sdl-config --cflags`
-CFLAGS	+= -O3 -ffast-math -fpeephole2 -fexpensive-optimizations
-#CFLAGS	+= -g
+CFLAGS	+= -Wall -pipe -I/usr/include $(shell sdl-config --cflags)
+CFLAGS2	+= -fpeephole2 -fexpensive-optimizations
+LDFLAGS +=
+
+TARGET   := fracturesandrust-by-atomimaitokahvi
+#variable de nettoyage
+RM_F := rm -f
+
+#compilation en paralelle si disponible
+ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ NUMJOBS = $(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+ MAKEFLAGS += -j$(NUMJOBS)
+endif
+
+export
 
 all: main script-buildbg script-filter script-buildcr script-loop script-addtime
 
 clean:
-	rm -f *.o
-	rm -f main script-buildbg script-filter script-buildcr script-loop script-addtime
+	-@$(RM_F) *.o
+	-@$(RM_F) main script-buildbg script-filter script-buildcr script-loop script-addtime
+	-@$(RM_F) $(TARGET)
 
 main: attract.o convolve.o filter.o input.o interpolate.o im.o list.o main.o pnm.o script.o util.o xmalloc.o tuherrus.o suodatin.o
-	$(CC) -o $@ $^ `sdl-config --libs` -lvorbisfile
+	$(CC) -o $(TARGET) $^ $(LDFLAGS) $(shell sdl-config --libs) -lvorbisfile -lm
 
 editor: convolve.o editor.o filter.o input.o interpolate.o im.o list.o pnm.o script.o util.o xmalloc.o
-	$(CC) -o $@ $^ `sdl-config --libs`
+	$(CC) -o $@ $^ $(LDFLAGS) $(shell sdl-config --libs)
 
 script-buildbg: script-buildbg.c
-	$(CC) -W -Wall -o $@ $^ -lm
+	$(CC) $(CFLAGS) -o $@ $^ -lm
 
 script-buildcr: script-buildcr.c
-	$(CC) -W -Wall -o $@ $^ -lm
+	$(CC) $(CFLAGS) -o $@ $^ -lm
 
 script-filter: script-filter.c
-	$(CC) -W -Wall -o $@ $^
+	$(CC) $(CFLAGS) -o $@ $^
 
 script-loop: script-loop.c
-	$(CC) -W -Wall -o $@ $^
+	$(CC) $(CFLAGS) -o $@ $^
 
 script-addtime: script-addtime.c
-	$(CC) -W -Wall -o $@ $^
+	$(CC) $(CFLAGS) -o $@ $^
 
 fixpgm: fixpgm.o pnm.o xmalloc.o im.o
-	$(CC) -o $@ $^
+	$(CC) $(CFLAGS) -o $@ $^
 
 %.o: %.c
-	$(CC) $(CFLAGS) -o $@ -c $<
+	$(CC) $(CFLAGS2) -o $@ -c $<
