--- a/main.c
+++ b/main.c
@@ -39,6 +39,13 @@
 #define DEMO_XASPECT 4.0f
 #define DEMO_YASPECT 3.0f
 
+void ToggleFullscreen(SDL_Window* Window) {
+	Uint32 FullscreenFlag = SDL_WINDOW_FULLSCREEN;
+	int IsFullscreen = SDL_GetWindowFlags(Window) & FullscreenFlag;
+	SDL_SetWindowFullscreen(Window, IsFullscreen ? 0 : FullscreenFlag);
+	SDL_ShowCursor(IsFullscreen);
+}
+
 
 int
 main(int argc, char *argv[])
@@ -214,6 +221,7 @@
 				switch (event.key.keysym.sym) {
 					case SDLK_ESCAPE:goto terminate; break;
 					case SDLK_q: goto terminate; break;
+					case SDLK_f: ToggleFullscreen(sdlWindow); break;
 				}
 			}
 			if (event.window.windowID == windowID)  {
@@ -267,6 +275,7 @@
 	return 0;
 }
 
+
 void
 write_pnm(Uint8 *pixels)
 {
