--- a/src/swamp_fly.c
+++ b/src/swamp_fly.c
@@ -1308,7 +1308,8 @@
 	}
 #endif
 
-	for(;;)
+	int quit = 0;
+	while(!quit)
 	{
 		// Perform drawing.
 #if defined(MANUAL_ADVANCE)
@@ -1345,7 +1346,7 @@
 		// Draw process.
 		if(currtick >= INTRO_LENGTH)
 		{
-			break;
+			quit++;
 		}
 		draw(currtick);
 		DDL_GL_SwapWindow(sdlWindow);
@@ -1354,38 +1355,18 @@
 		tick_cumulative += FRAME_TICKS;
 #endif
 
-		// Abort on any keypress.
-		if(DDL_PollEvent(&event) && (event.type == SDL_KEYDOWN))
+		while(SDL_PollEvent(&event))
 		{
-#if defined(USE_LD)
-			switch(event.key.keysym.sym)
-			{
-#if defined(MANUAL_ADVANCE)
-			case SDLK_RSHIFT:
-				currtickmul = 100;
-				break;
-			case SDLK_LSHIFT:
-				currtickmul = 1;
-				break;
-			case SDLK_PERIOD:
-				currtick += 20 * currtickmul;
-				break;
-			case SDLK_COMMA:
-				currtick -= 20 * currtickmul;
-				break;
-#endif
-			case SDLK_SPACE:
-				printf("Ticks: %i\n", currtick);
-			default:
-				break;
+			if (event.type == SDL_QUIT) {
+				quit++;
 			}
-			if(event.key.keysym.sym == SDLK_ESCAPE)
-			{
-				break;
+			if (event.type == SDL_KEYUP) {
+				switch (event.key.keysym.sym) {
+				case SDLK_SPACE:printf("Ticks: %i\n", currtick); break;
+				case SDLK_ESCAPE: quit++; break;
+				case SDLK_q: quit++; break;
+				}
 			}
-#else
-			break;
-#endif
 		}
 	}
 
