--- a/App/usr/share/applications/xppentablet.desktop
+++ b/App/usr/share/applications/xppentablet.desktop
@@ -1,11 +1,13 @@
 [Desktop Entry]
 Name=xppentablet
 Comment=XPPen driver
-Exec=/usr/lib/pentablet/PenTablet.sh
+Comment[fr_FR]=Pilote XPPen
+Exec=xppenutils
 Icon=/usr/share/icons/hicolor/256x256/apps/xppentablet.png
 Terminal=false
 Type=Application
 X-Ubuntu-Touch=true                                        
 Categories=Application;Utility;
 StartupNotify=true
-Name[en_US]=xppentablet
+Name[en_US]=XP-Pen Tablet
+Name[fr_FR]=Tablette XP-Pen
