--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,8 @@
 #variables de compilation
 SHELL := /bin/sh
 CC := gcc
-CFLAGS += -Wall -pipe $(shell sdl-config --cflags) $(shell pkg-config gl --cflags) -std=gnu90
-LDFLAGS += $(shell sdl-config --libs) $(shell pkg-config gl --libs) -lm
+CFLAGS += -Wall -pipe $(shell pkg-config gl sdl2 --cflags) -std=gnu90
+LDFLAGS += $(shell pkg-config gl sdl2 --libs) -lm
 MAKE := make
 AR := ar
 RANLIB := ranlib
--- a/gl_render.c
+++ b/gl_render.c
@@ -5,7 +5,7 @@
 #include "wavefront.h"
 #include "gl_render.h"
 
-#include <SDL_opengl.h>
+#include <SDL2/SDL_opengl.h>
 
 static GLint lname[8]={GL_LIGHT0,GL_LIGHT1,GL_LIGHT2,GL_LIGHT3,
 	               GL_LIGHT4,GL_LIGHT5,GL_LIGHT6,GL_LIGHT7};
--- a/main.c
+++ b/main.c
@@ -2,8 +2,8 @@
     A quick & dirty 64k for Ass'06
  */
 
-#include <SDL.h>
-#include <SDL_opengl.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_opengl.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <unistd.h>
@@ -53,12 +53,17 @@
 	mzx_init();
 	mzx_get(motti);
 
-	if(argc>=2 || argv[0][strlen(argv[0])-1]=='w')
-		SDL_SetVideoMode(xs,ys,32,SDL_OPENGL);
-	else
-		SDL_SetVideoMode(xs,ys,32,SDL_OPENGL+SDL_FULLSCREEN);
-	SDL_ShowCursor(0);
-	SDL_WM_SetCaption("Emo Defect by The Fiture Crew",0);
+	SDL_Window *window;
+
+	if(argc>=2 || argv[0][strlen(argv[0])-1]=='w') {
+		window = SDL_CreateWindow("Emo Defect by The Fiture Crew",SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED,xs,ys,SDL_WINDOW_OPENGL);
+		SDL_ShowCursor(1);
+	}
+	else {
+		window = SDL_CreateWindow("Emo Defect by The Fiture Crew",SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED,xs,ys, SDL_WINDOW_RESIZABLE | SDL_WINDOW_OPENGL | SDL_WINDOW_FULLSCREEN);
+		SDL_ShowCursor(0);
+	}
+	SDL_GLContext glcontext = SDL_GL_CreateContext(window);
 
 	load_dump(&s[0],"kuutio.dump");
 	load_dump(&s[1],"pallo.dump");
@@ -301,12 +306,14 @@
 			glPopAttrib();
 		}
 
-		SDL_GL_SwapBuffers();
+		SDL_GL_SwapWindow(window);
 
 		if(modpos>=1848)
 			quit++;
 	}
 
 	mzx_close();
+	SDL_GL_DeleteContext(glcontext);
+	SDL_DestroyWindow(window);
 	SDL_Quit();
 }
--- a/cool_mzx/cool_mzx.c
+++ b/cool_mzx/cool_mzx.c
@@ -12,7 +12,7 @@
 #include "cool_mzx.h"
 #include "sd_main.h"
 #include "pt.h"
-#include <SDL/SDL.h>
+#include <SDL2/SDL.h>
 
 extern int pt_curperiod[32],           // kanavan nykyinen periodi
            pt_curnote[32],             // kanavan nykyinen sävel
@@ -198,7 +198,7 @@
 	if(!laite)
 	{
 		SDL_Init(SDL_INIT_TIMER);
-		SDL_SetTimer(jacaja, timeri);
+		SDL_AddTimer(jacaja, timeri,NULL);
 		return;
 	}
 	SDL_PauseAudio(0);
@@ -280,7 +280,7 @@
 {
 	if(!laite)
 	{
-		SDL_SetTimer(0,NULL);
+		SDL_AddTimer(0,NULL,NULL);
 		return;
 	}
 
--- a/cool_mzx/sd_dsp16.c
+++ b/cool_mzx/sd_dsp16.c
@@ -9,7 +9,7 @@
          2.9.2002: Some minor cleaning for SDL
  */
 
-#include <SDL/SDL.h>
+#include <SDL2/SDL.h>
 #include <stdio.h>
 #include <stdlib.h>
 
