--- a/main.c
+++ b/main.c
@@ -87,11 +87,17 @@
 
 	while(!quit)
 	{
-		while(SDL_PollEvent(&e)>0)
+		while(SDL_PollEvent(&e))
 		{
-			if(e.type==SDL_KEYDOWN)
-				if(e.key.keysym.sym==SDLK_ESCAPE)
-					quit++;
+			if (e.type == SDL_QUIT) {
+				quit++;
+			}
+			if (e.type == SDL_KEYUP) {
+				switch (e.key.keysym.sym) {
+					case SDLK_ESCAPE: quit++; break;
+					case SDLK_q: quit++; break;
+				}
+			}
 		}
 		modpos=mzx_position();
 
