--- a/anrufausschweden-by-unik
+++ b/anrufausschweden-by-unik
@@ -1,6 +1,6 @@
 #!/bin/bash -e
 
-source common-install-function
+source /usr/share/common-install-function-ddebs/common-install-function
 
 domain=anrufausschweden-by-unik
 
@@ -11,6 +11,7 @@
 function run()
 {
 	NSH_CreateTmpDomainDir
+	NSH_SHAREDIR=/usr/lib/i386-linux-gnu/
 	NSH_CopyFileInTmp ikea
 	/tmp/$domain/ikea
 	NSH_RemoveTmpDomainDir
